Synthesis failed for nodejs-firestore-session

Hello! Autosynth couldn't regenerate nodejs-firestore-session. :broken_heart:

Here's the output from running `synth.py`:

```
htool --metadata synth.metadata synth.py --
2020-11-10 02:54:55,830 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/nodejs-firestore-session/synth.py.
On branch autosynth-6
nothing to commit, working tree clean
2020-11-10 02:54:55,965 synthtool [DEBUG] > Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
DEBUG:synthtool:Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
.eslintignore
.eslintrc.json
.gitattributes
.github/ISSUE_TEMPLATE/bug_report.md
.github/ISSUE_TEMPLATE/feature_request.md
.github/ISSUE_TEMPLATE/support_request.md
.github/PULL_REQUEST_TEMPLATE.md
.github/release-please.yml
.github/workflows/ci.yaml
.kokoro/.gitattributes
.kokoro/common.cfg
.kokoro/continuous/node10/common.cfg
.kokoro/continuous/node10/docs.cfg
.kokoro/continuous/node10/test.cfg
.kokoro/continuous/node12/common.cfg
.kokoro/continuous/node12/lint.cfg
.kokoro/continuous/node12/samples-test.cfg
.kokoro/continuous/node12/system-test.cfg
.kokoro/continuous/node12/test.cfg
.kokoro/docs.sh
.kokoro/lint.sh
.kokoro/populate-secrets.sh
.kokoro/presubmit/node10/common.cfg
.kokoro/presubmit/node12/common.cfg
.kokoro/presubmit/node12/samples-test.cfg
.kokoro/presubmit/node12/system-test.cfg
.kokoro/presubmit/node12/test.cfg
.kokoro/publish.sh
.kokoro/release/docs-devsite.cfg
.kokoro/release/docs-devsite.sh
.kokoro/release/docs.cfg
.kokoro/release/docs.sh
.kokoro/release/publish.cfg
.kokoro/samples-test.sh
.kokoro/system-test.sh
.kokoro/test.bat
.kokoro/test.sh
.kokoro/trampoline.sh
.kokoro/trampoline_v2.sh
.mocharc.js
.nycrc
.prettierignore
.prettierrc.js
.trampolinerc
CODE_OF_CONDUCT.md
CONTRIBUTING.md
LICENSE
README.md
api-extractor.json
renovate.json
samples/README.md
2020-11-10 02:54:56,139 synthtool [DEBUG] > Installing dependencies...
DEBUG:synthtool:Installing dependencies...
npm WARN deprecated ts-simple-ast@12.4.0: NOTICE: ts-simple-ast has been renamed to ts-morph and version reset to 1.0.0. Switch at your leisure...
npm WARN deprecated chokidar@2.1.8: Chokidar 2 will break on node v14+. Upgrade to chokidar 3 with 15x less dependencies.
npm WARN deprecated opn@6.0.0: The package has been renamed to `open`
npm WARN deprecated fsevents@1.2.13: fsevents 1 will break on node v14+ and could be using insecure binaries. Upgrade to fsevents 2.
npm WARN deprecated core-js@2.6.11: core-js@<3 is no longer maintained and not recommended for usage due to the number of issues. Please, upgrade your dependencies to the actual version of core-js@3.
npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated

> core-js@2.6.11 postinstall /home/kbuilder/.cache/synthtool/nodejs-firestore-session/node_modules/core-js
> node -e "try{require('./postinstall')}catch(e){}"


> protobufjs@6.10.1 postinstall /home/kbuilder/.cache/synthtool/nodejs-firestore-session/node_modules/protobufjs
> node scripts/postinstall


> @compodoc/compodoc@1.1.11 postinstall /home/kbuilder/.cache/synthtool/nodejs-firestore-session/node_modules/@compodoc/compodoc
> opencollective-postinstall || exit 0

[96m[1mThank you for using @compodoc/compodoc![96m[1m
[0m[96mIf you rely on this package, please consider supporting our open collective:[22m[39m
> [94mhttps://opencollective.com/compodoc/donate[0m


> @google-cloud/connect-firestore@2.0.0 prepare /home/kbuilder/.cache/synthtool/nodejs-firestore-session
> npm run compile


> @google-cloud/connect-firestore@2.0.0 precompile /home/kbuilder/.cache/synthtool/nodejs-firestore-session
> gts clean

version: 14
Removing build ...

> @google-cloud/connect-firestore@2.0.0 compile /home/kbuilder/.cache/synthtool/nodejs-firestore-session
> tsc -p .

[96msrc/index.ts[0m:[93m27[0m:[93m11[0m - [91merror[0m[90m TS2559: [0mType 'StoreOptions' has no properties in common with type 'EventEmitterOptions'.

[7m27[0m     super(options || {});
[7m  [0m [91m          ~~~~~~~~~~~~~[0m

[96msrc/index.ts[0m:[93m37[0m:[93m54[0m - [91merror[0m[90m TS2694: [0mNamespace 'global.Express' has no exported member 'SessionData'.

[7m37[0m     callback: (err?: Error | null, session?: Express.SessionData) => void
[7m  [0m [91m                                                     ~~~~~~~~~~~[0m

[96msrc/index.ts[0m:[93m59[0m:[93m22[0m - [91merror[0m[90m TS2694: [0mNamespace 'global.Express' has no exported member 'SessionData'.

[7m59[0m     session: Express.SessionData,
[7m  [0m [91m                     ~~~~~~~~~~~[0m

[96msystem-test/session.ts[0m:[93m34[0m:[93m57[0m - [91merror[0m[90m TS2694: [0mNamespace 'global.Express' has no exported member 'SessionData'.

[7m34[0m     const sessionData = ({foo: 'bar'} as {}) as Express.SessionData;
[7m  [0m [91m                                                        ~~~~~~~~~~~[0m

[96mtest/test.ts[0m:[93m99[0m:[93m42[0m - [91merror[0m[90m TS2694: [0mNamespace 'global.Express' has no exported member 'SessionData'.

[7m99[0m     store.set(expectedSid, {} as Express.SessionData, assert.ifError);
[7m  [0m [91m                                         ~~~~~~~~~~~[0m


Found 5 errors.

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @google-cloud/connect-firestore@2.0.0 compile: `tsc -p .`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @google-cloud/connect-firestore@2.0.0 compile script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/kbuilder/.npm/_logs/2020-11-10T10_55_20_985Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @google-cloud/connect-firestore@2.0.0 prepare: `npm run compile`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @google-cloud/connect-firestore@2.0.0 prepare script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/kbuilder/.npm/_logs/2020-11-10T10_55_21_055Z-debug.log
2020-11-10 02:55:21,087 synthtool [ERROR] > Failed executing npm install:

None
ERROR:synthtool:Failed executing npm install:

None
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/nodejs-firestore-session/synth.py", line 15, in <module>
    node.install()
  File "/tmpfs/src/github/synthtool/synthtool/languages/node.py", line 167, in install
    shell.run(["npm", "install"], hide_output=hide_output)
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 39, in run
    raise exc
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 33, in run
    encoding="utf-8",
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['npm', 'install']' returned non-zero exit status 1.
2020-11-10 02:55:21,184 autosynth [ERROR] > Synthesis failed
2020-11-10 02:55:21,184 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at b623c57 build(node): add KOKORO_BUILD_ARTIFACTS_SUBDIR to env (#147)
2020-11-10 02:55:21,192 autosynth [DEBUG] > Running: git checkout autosynth
Switched to branch 'autosynth'
2020-11-10 02:55:21,197 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Removing node_modules/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 354, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 189, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 334, in _inner_main
    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 65, in synthesize_loop
    has_changes = toolbox.synthesize_version_in_new_branch(synthesizer, youngest)
  File "/tmpfs/src/github/synthtool/autosynth/synth_toolbox.py", line 259, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/5a04e869-da9f-4fb2-803d-732322e5bde5/targets/github%2Fsynthtool;config=default/tests;query=nodejs-firestore-session;failed=false).

Autosynth is still having trouble generating nodejs-firestore-session. :sob:

Here's the output from running `synth.py`:

```
s
.github/ISSUE_TEMPLATE/bug_report.md
.github/ISSUE_TEMPLATE/feature_request.md
.github/ISSUE_TEMPLATE/support_request.md
.github/PULL_REQUEST_TEMPLATE.md
.github/release-please.yml
.github/workflows/ci.yaml
.kokoro/.gitattributes
.kokoro/common.cfg
.kokoro/continuous/node10/common.cfg
.kokoro/continuous/node10/docs.cfg
.kokoro/continuous/node10/test.cfg
.kokoro/continuous/node12/common.cfg
.kokoro/continuous/node12/lint.cfg
.kokoro/continuous/node12/samples-test.cfg
.kokoro/continuous/node12/system-test.cfg
.kokoro/continuous/node12/test.cfg
.kokoro/docs.sh
.kokoro/lint.sh
.kokoro/populate-secrets.sh
.kokoro/presubmit/node10/common.cfg
.kokoro/presubmit/node12/common.cfg
.kokoro/presubmit/node12/samples-test.cfg
.kokoro/presubmit/node12/system-test.cfg
.kokoro/presubmit/node12/test.cfg
.kokoro/publish.sh
.kokoro/release/docs-devsite.cfg
.kokoro/release/docs-devsite.sh
.kokoro/release/docs.cfg
.kokoro/release/docs.sh
.kokoro/release/publish.cfg
.kokoro/samples-test.sh
.kokoro/system-test.sh
.kokoro/test.bat
.kokoro/test.sh
.kokoro/trampoline.sh
.kokoro/trampoline_v2.sh
.mocharc.js
.nycrc
.prettierignore
.prettierrc.js
.trampolinerc
CODE_OF_CONDUCT.md
CONTRIBUTING.md
LICENSE
README.md
api-extractor.json
renovate.json
samples/README.md
2020-11-11 02:50:57,415 synthtool [DEBUG] > Installing dependencies...
DEBUG:synthtool:Installing dependencies...
npm WARN deprecated ts-simple-ast@12.4.0: NOTICE: ts-simple-ast has been renamed to ts-morph and version reset to 1.0.0. Switch at your leisure...
npm WARN deprecated chokidar@2.1.8: Chokidar 2 will break on node v14+. Upgrade to chokidar 3 with 15x less dependencies.
npm WARN deprecated opn@6.0.0: The package has been renamed to `open`
npm WARN deprecated fsevents@1.2.13: fsevents 1 will break on node v14+ and could be using insecure binaries. Upgrade to fsevents 2.
npm WARN deprecated core-js@2.6.11: core-js@<3 is no longer maintained and not recommended for usage due to the number of issues. Please, upgrade your dependencies to the actual version of core-js@3.
npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated

> core-js@2.6.11 postinstall /home/kbuilder/.cache/synthtool/nodejs-firestore-session/node_modules/core-js
> node -e "try{require('./postinstall')}catch(e){}"

[96mThank you for using core-js ([94m https://github.com/zloirock/core-js [96m) for polyfilling JavaScript standard library![0m

[96mThe project needs your help! Please consider supporting of core-js on Open Collective or Patreon: [0m
[96m>[94m https://opencollective.com/core-js [0m
[96m>[94m https://www.patreon.com/zloirock [0m

[96mAlso, the author of core-js ([94m https://github.com/zloirock [96m) is looking for a good job -)[0m


> protobufjs@6.10.1 postinstall /home/kbuilder/.cache/synthtool/nodejs-firestore-session/node_modules/protobufjs
> node scripts/postinstall


> @compodoc/compodoc@1.1.11 postinstall /home/kbuilder/.cache/synthtool/nodejs-firestore-session/node_modules/@compodoc/compodoc
> opencollective-postinstall || exit 0

[96m[1mThank you for using @compodoc/compodoc![96m[1m
[0m[96mIf you rely on this package, please consider supporting our open collective:[22m[39m
> [94mhttps://opencollective.com/compodoc/donate[0m


> @google-cloud/connect-firestore@2.0.0 prepare /home/kbuilder/.cache/synthtool/nodejs-firestore-session
> npm run compile


> @google-cloud/connect-firestore@2.0.0 precompile /home/kbuilder/.cache/synthtool/nodejs-firestore-session
> gts clean

version: 14
Removing build ...

> @google-cloud/connect-firestore@2.0.0 compile /home/kbuilder/.cache/synthtool/nodejs-firestore-session
> tsc -p .

[96msrc/index.ts[0m:[93m27[0m:[93m11[0m - [91merror[0m[90m TS2559: [0mType 'StoreOptions' has no properties in common with type 'EventEmitterOptions'.

[7m27[0m     super(options || {});
[7m  [0m [91m          ~~~~~~~~~~~~~[0m

[96msrc/index.ts[0m:[93m37[0m:[93m54[0m - [91merror[0m[90m TS2694: [0mNamespace 'global.Express' has no exported member 'SessionData'.

[7m37[0m     callback: (err?: Error | null, session?: Express.SessionData) => void
[7m  [0m [91m                                                     ~~~~~~~~~~~[0m

[96msrc/index.ts[0m:[93m59[0m:[93m22[0m - [91merror[0m[90m TS2694: [0mNamespace 'global.Express' has no exported member 'SessionData'.

[7m59[0m     session: Express.SessionData,
[7m  [0m [91m                     ~~~~~~~~~~~[0m

[96msystem-test/session.ts[0m:[93m34[0m:[93m57[0m - [91merror[0m[90m TS2694: [0mNamespace 'global.Express' has no exported member 'SessionData'.

[7m34[0m     const sessionData = ({foo: 'bar'} as {}) as Express.SessionData;
[7m  [0m [91m                                                        ~~~~~~~~~~~[0m

[96mtest/test.ts[0m:[93m99[0m:[93m42[0m - [91merror[0m[90m TS2694: [0mNamespace 'global.Express' has no exported member 'SessionData'.

[7m99[0m     store.set(expectedSid, {} as Express.SessionData, assert.ifError);
[7m  [0m [91m                                         ~~~~~~~~~~~[0m


Found 5 errors.

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @google-cloud/connect-firestore@2.0.0 compile: `tsc -p .`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @google-cloud/connect-firestore@2.0.0 compile script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/kbuilder/.npm/_logs/2020-11-11T10_51_21_401Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @google-cloud/connect-firestore@2.0.0 prepare: `npm run compile`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @google-cloud/connect-firestore@2.0.0 prepare script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/kbuilder/.npm/_logs/2020-11-11T10_51_21_491Z-debug.log
2020-11-11 02:51:21,541 synthtool [ERROR] > Failed executing npm install:

None
ERROR:synthtool:Failed executing npm install:

None
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/nodejs-firestore-session/synth.py", line 15, in <module>
    node.install()
  File "/tmpfs/src/github/synthtool/synthtool/languages/node.py", line 167, in install
    shell.run(["npm", "install"], hide_output=hide_output)
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 39, in run
    raise exc
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 33, in run
    encoding="utf-8",
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['npm', 'install']' returned non-zero exit status 1.
2020-11-11 02:51:21,665 autosynth [ERROR] > Synthesis failed
2020-11-11 02:51:21,665 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at b623c57 build(node): add KOKORO_BUILD_ARTIFACTS_SUBDIR to env (#147)
2020-11-11 02:51:21,672 autosynth [DEBUG] > Running: git checkout autosynth
Switched to branch 'autosynth'
2020-11-11 02:51:21,677 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Removing node_modules/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 354, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 189, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 334, in _inner_main
    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 65, in synthesize_loop
    has_changes = toolbox.synthesize_version_in_new_branch(synthesizer, youngest)
  File "/tmpfs/src/github/synthtool/autosynth/synth_toolbox.py", line 259, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/220301f2-3013-4f2a-8df5-9e471bc8aaf3/targets/github%2Fsynthtool;config=default/tests;query=nodejs-firestore-session;failed=false).

Autosynth is still having trouble generating nodejs-firestore-session. :sob:

Here's the output from running `synth.py`:

```
s
.github/ISSUE_TEMPLATE/bug_report.md
.github/ISSUE_TEMPLATE/feature_request.md
.github/ISSUE_TEMPLATE/support_request.md
.github/PULL_REQUEST_TEMPLATE.md
.github/release-please.yml
.github/workflows/ci.yaml
.kokoro/.gitattributes
.kokoro/common.cfg
.kokoro/continuous/node10/common.cfg
.kokoro/continuous/node10/docs.cfg
.kokoro/continuous/node10/test.cfg
.kokoro/continuous/node12/common.cfg
.kokoro/continuous/node12/lint.cfg
.kokoro/continuous/node12/samples-test.cfg
.kokoro/continuous/node12/system-test.cfg
.kokoro/continuous/node12/test.cfg
.kokoro/docs.sh
.kokoro/lint.sh
.kokoro/populate-secrets.sh
.kokoro/presubmit/node10/common.cfg
.kokoro/presubmit/node12/common.cfg
.kokoro/presubmit/node12/samples-test.cfg
.kokoro/presubmit/node12/system-test.cfg
.kokoro/presubmit/node12/test.cfg
.kokoro/publish.sh
.kokoro/release/docs-devsite.cfg
.kokoro/release/docs-devsite.sh
.kokoro/release/docs.cfg
.kokoro/release/docs.sh
.kokoro/release/publish.cfg
.kokoro/samples-test.sh
.kokoro/system-test.sh
.kokoro/test.bat
.kokoro/test.sh
.kokoro/trampoline.sh
.kokoro/trampoline_v2.sh
.mocharc.js
.nycrc
.prettierignore
.prettierrc.js
.trampolinerc
CODE_OF_CONDUCT.md
CONTRIBUTING.md
LICENSE
README.md
api-extractor.json
renovate.json
samples/README.md
2020-11-12 02:50:44,051 synthtool [DEBUG] > Installing dependencies...
DEBUG:synthtool:Installing dependencies...
npm WARN deprecated ts-simple-ast@12.4.0: NOTICE: ts-simple-ast has been renamed to ts-morph and version reset to 1.0.0. Switch at your leisure...
npm WARN deprecated chokidar@2.1.8: Chokidar 2 will break on node v14+. Upgrade to chokidar 3 with 15x less dependencies.
npm WARN deprecated opn@6.0.0: The package has been renamed to `open`
npm WARN deprecated fsevents@1.2.13: fsevents 1 will break on node v14+ and could be using insecure binaries. Upgrade to fsevents 2.
npm WARN deprecated core-js@2.6.11: core-js@<3 is no longer maintained and not recommended for usage due to the number of issues. Please, upgrade your dependencies to the actual version of core-js@3.
npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated

> core-js@2.6.11 postinstall /home/kbuilder/.cache/synthtool/nodejs-firestore-session/node_modules/core-js
> node -e "try{require('./postinstall')}catch(e){}"

[96mThank you for using core-js ([94m https://github.com/zloirock/core-js [96m) for polyfilling JavaScript standard library![0m

[96mThe project needs your help! Please consider supporting of core-js on Open Collective or Patreon: [0m
[96m>[94m https://opencollective.com/core-js [0m
[96m>[94m https://www.patreon.com/zloirock [0m

[96mAlso, the author of core-js ([94m https://github.com/zloirock [96m) is looking for a good job -)[0m


> protobufjs@6.10.1 postinstall /home/kbuilder/.cache/synthtool/nodejs-firestore-session/node_modules/protobufjs
> node scripts/postinstall


> @compodoc/compodoc@1.1.11 postinstall /home/kbuilder/.cache/synthtool/nodejs-firestore-session/node_modules/@compodoc/compodoc
> opencollective-postinstall || exit 0

[96m[1mThank you for using @compodoc/compodoc![96m[1m
[0m[96mIf you rely on this package, please consider supporting our open collective:[22m[39m
> [94mhttps://opencollective.com/compodoc/donate[0m


> @google-cloud/connect-firestore@2.0.0 prepare /home/kbuilder/.cache/synthtool/nodejs-firestore-session
> npm run compile


> @google-cloud/connect-firestore@2.0.0 precompile /home/kbuilder/.cache/synthtool/nodejs-firestore-session
> gts clean

version: 14
Removing build ...

> @google-cloud/connect-firestore@2.0.0 compile /home/kbuilder/.cache/synthtool/nodejs-firestore-session
> tsc -p .

[96msrc/index.ts[0m:[93m27[0m:[93m11[0m - [91merror[0m[90m TS2559: [0mType 'StoreOptions' has no properties in common with type 'EventEmitterOptions'.

[7m27[0m     super(options || {});
[7m  [0m [91m          ~~~~~~~~~~~~~[0m

[96msrc/index.ts[0m:[93m37[0m:[93m54[0m - [91merror[0m[90m TS2694: [0mNamespace 'global.Express' has no exported member 'SessionData'.

[7m37[0m     callback: (err?: Error | null, session?: Express.SessionData) => void
[7m  [0m [91m                                                     ~~~~~~~~~~~[0m

[96msrc/index.ts[0m:[93m59[0m:[93m22[0m - [91merror[0m[90m TS2694: [0mNamespace 'global.Express' has no exported member 'SessionData'.

[7m59[0m     session: Express.SessionData,
[7m  [0m [91m                     ~~~~~~~~~~~[0m

[96msystem-test/session.ts[0m:[93m34[0m:[93m57[0m - [91merror[0m[90m TS2694: [0mNamespace 'global.Express' has no exported member 'SessionData'.

[7m34[0m     const sessionData = ({foo: 'bar'} as {}) as Express.SessionData;
[7m  [0m [91m                                                        ~~~~~~~~~~~[0m

[96mtest/test.ts[0m:[93m99[0m:[93m42[0m - [91merror[0m[90m TS2694: [0mNamespace 'global.Express' has no exported member 'SessionData'.

[7m99[0m     store.set(expectedSid, {} as Express.SessionData, assert.ifError);
[7m  [0m [91m                                         ~~~~~~~~~~~[0m


Found 5 errors.

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @google-cloud/connect-firestore@2.0.0 compile: `tsc -p .`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @google-cloud/connect-firestore@2.0.0 compile script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/kbuilder/.npm/_logs/2020-11-12T10_51_07_931Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @google-cloud/connect-firestore@2.0.0 prepare: `npm run compile`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @google-cloud/connect-firestore@2.0.0 prepare script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/kbuilder/.npm/_logs/2020-11-12T10_51_08_027Z-debug.log
2020-11-12 02:51:08,064 synthtool [ERROR] > Failed executing npm install:

None
ERROR:synthtool:Failed executing npm install:

None
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/nodejs-firestore-session/synth.py", line 15, in <module>
    node.install()
  File "/tmpfs/src/github/synthtool/synthtool/languages/node.py", line 167, in install
    shell.run(["npm", "install"], hide_output=hide_output)
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 39, in run
    raise exc
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 33, in run
    encoding="utf-8",
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['npm', 'install']' returned non-zero exit status 1.
2020-11-12 02:51:08,162 autosynth [ERROR] > Synthesis failed
2020-11-12 02:51:08,162 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at b623c57 build(node): add KOKORO_BUILD_ARTIFACTS_SUBDIR to env (#147)
2020-11-12 02:51:08,170 autosynth [DEBUG] > Running: git checkout autosynth
Switched to branch 'autosynth'
2020-11-12 02:51:08,175 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Removing node_modules/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 354, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 189, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 334, in _inner_main
    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 65, in synthesize_loop
    has_changes = toolbox.synthesize_version_in_new_branch(synthesizer, youngest)
  File "/tmpfs/src/github/synthtool/autosynth/synth_toolbox.py", line 259, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/985151cf-6dc4-4547-8bf8-14b00137c89f/targets/github%2Fsynthtool;config=default/tests;query=nodejs-firestore-session;failed=false).

Autosynth is still having trouble generating nodejs-firestore-session. :sob:

Here's the output from running `synth.py`:

```
s
.github/ISSUE_TEMPLATE/bug_report.md
.github/ISSUE_TEMPLATE/feature_request.md
.github/ISSUE_TEMPLATE/support_request.md
.github/PULL_REQUEST_TEMPLATE.md
.github/release-please.yml
.github/workflows/ci.yaml
.kokoro/.gitattributes
.kokoro/common.cfg
.kokoro/continuous/node10/common.cfg
.kokoro/continuous/node10/docs.cfg
.kokoro/continuous/node10/test.cfg
.kokoro/continuous/node12/common.cfg
.kokoro/continuous/node12/lint.cfg
.kokoro/continuous/node12/samples-test.cfg
.kokoro/continuous/node12/system-test.cfg
.kokoro/continuous/node12/test.cfg
.kokoro/docs.sh
.kokoro/lint.sh
.kokoro/populate-secrets.sh
.kokoro/presubmit/node10/common.cfg
.kokoro/presubmit/node12/common.cfg
.kokoro/presubmit/node12/samples-test.cfg
.kokoro/presubmit/node12/system-test.cfg
.kokoro/presubmit/node12/test.cfg
.kokoro/publish.sh
.kokoro/release/docs-devsite.cfg
.kokoro/release/docs-devsite.sh
.kokoro/release/docs.cfg
.kokoro/release/docs.sh
.kokoro/release/publish.cfg
.kokoro/samples-test.sh
.kokoro/system-test.sh
.kokoro/test.bat
.kokoro/test.sh
.kokoro/trampoline.sh
.kokoro/trampoline_v2.sh
.mocharc.js
.nycrc
.prettierignore
.prettierrc.js
.trampolinerc
CODE_OF_CONDUCT.md
CONTRIBUTING.md
LICENSE
README.md
api-extractor.json
renovate.json
samples/README.md
2020-11-13 03:00:33,333 synthtool [DEBUG] > Installing dependencies...
DEBUG:synthtool:Installing dependencies...
npm WARN deprecated ts-simple-ast@12.4.0: NOTICE: ts-simple-ast has been renamed to ts-morph and version reset to 1.0.0. Switch at your leisure...
npm WARN deprecated chokidar@2.1.8: Chokidar 2 will break on node v14+. Upgrade to chokidar 3 with 15x less dependencies.
npm WARN deprecated opn@6.0.0: The package has been renamed to `open`
npm WARN deprecated fsevents@1.2.13: fsevents 1 will break on node v14+ and could be using insecure binaries. Upgrade to fsevents 2.
npm WARN deprecated core-js@2.6.11: core-js@<3 is no longer maintained and not recommended for usage due to the number of issues. Please, upgrade your dependencies to the actual version of core-js@3.
npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated

> core-js@2.6.11 postinstall /home/kbuilder/.cache/synthtool/nodejs-firestore-session/node_modules/core-js
> node -e "try{require('./postinstall')}catch(e){}"

[96mThank you for using core-js ([94m https://github.com/zloirock/core-js [96m) for polyfilling JavaScript standard library![0m

[96mThe project needs your help! Please consider supporting of core-js on Open Collective or Patreon: [0m
[96m>[94m https://opencollective.com/core-js [0m
[96m>[94m https://www.patreon.com/zloirock [0m

[96mAlso, the author of core-js ([94m https://github.com/zloirock [96m) is looking for a good job -)[0m


> protobufjs@6.10.1 postinstall /home/kbuilder/.cache/synthtool/nodejs-firestore-session/node_modules/protobufjs
> node scripts/postinstall


> @compodoc/compodoc@1.1.11 postinstall /home/kbuilder/.cache/synthtool/nodejs-firestore-session/node_modules/@compodoc/compodoc
> opencollective-postinstall || exit 0

[96m[1mThank you for using @compodoc/compodoc![96m[1m
[0m[96mIf you rely on this package, please consider supporting our open collective:[22m[39m
> [94mhttps://opencollective.com/compodoc/donate[0m


> @google-cloud/connect-firestore@2.0.0 prepare /home/kbuilder/.cache/synthtool/nodejs-firestore-session
> npm run compile


> @google-cloud/connect-firestore@2.0.0 precompile /home/kbuilder/.cache/synthtool/nodejs-firestore-session
> gts clean

version: 14
Removing build ...

> @google-cloud/connect-firestore@2.0.0 compile /home/kbuilder/.cache/synthtool/nodejs-firestore-session
> tsc -p .

[96msrc/index.ts[0m:[93m27[0m:[93m11[0m - [91merror[0m[90m TS2559: [0mType 'StoreOptions' has no properties in common with type 'EventEmitterOptions'.

[7m27[0m     super(options || {});
[7m  [0m [91m          ~~~~~~~~~~~~~[0m

[96msrc/index.ts[0m:[93m37[0m:[93m54[0m - [91merror[0m[90m TS2694: [0mNamespace 'global.Express' has no exported member 'SessionData'.

[7m37[0m     callback: (err?: Error | null, session?: Express.SessionData) => void
[7m  [0m [91m                                                     ~~~~~~~~~~~[0m

[96msrc/index.ts[0m:[93m59[0m:[93m22[0m - [91merror[0m[90m TS2694: [0mNamespace 'global.Express' has no exported member 'SessionData'.

[7m59[0m     session: Express.SessionData,
[7m  [0m [91m                     ~~~~~~~~~~~[0m

[96msystem-test/session.ts[0m:[93m34[0m:[93m57[0m - [91merror[0m[90m TS2694: [0mNamespace 'global.Express' has no exported member 'SessionData'.

[7m34[0m     const sessionData = ({foo: 'bar'} as {}) as Express.SessionData;
[7m  [0m [91m                                                        ~~~~~~~~~~~[0m

[96mtest/test.ts[0m:[93m99[0m:[93m42[0m - [91merror[0m[90m TS2694: [0mNamespace 'global.Express' has no exported member 'SessionData'.

[7m99[0m     store.set(expectedSid, {} as Express.SessionData, assert.ifError);
[7m  [0m [91m                                         ~~~~~~~~~~~[0m


Found 5 errors.

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @google-cloud/connect-firestore@2.0.0 compile: `tsc -p .`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @google-cloud/connect-firestore@2.0.0 compile script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/kbuilder/.npm/_logs/2020-11-13T11_00_57_134Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @google-cloud/connect-firestore@2.0.0 prepare: `npm run compile`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @google-cloud/connect-firestore@2.0.0 prepare script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/kbuilder/.npm/_logs/2020-11-13T11_00_57_207Z-debug.log
2020-11-13 03:00:57,255 synthtool [ERROR] > Failed executing npm install:

None
ERROR:synthtool:Failed executing npm install:

None
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/nodejs-firestore-session/synth.py", line 15, in <module>
    node.install()
  File "/tmpfs/src/github/synthtool/synthtool/languages/node.py", line 167, in install
    shell.run(["npm", "install"], hide_output=hide_output)
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 39, in run
    raise exc
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 33, in run
    encoding="utf-8",
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['npm', 'install']' returned non-zero exit status 1.
2020-11-13 03:00:57,362 autosynth [ERROR] > Synthesis failed
2020-11-13 03:00:57,362 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at b623c57 build(node): add KOKORO_BUILD_ARTIFACTS_SUBDIR to env (#147)
2020-11-13 03:00:57,370 autosynth [DEBUG] > Running: git checkout autosynth
Switched to branch 'autosynth'
2020-11-13 03:00:57,375 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Removing node_modules/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 354, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 189, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 334, in _inner_main
    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 65, in synthesize_loop
    has_changes = toolbox.synthesize_version_in_new_branch(synthesizer, youngest)
  File "/tmpfs/src/github/synthtool/autosynth/synth_toolbox.py", line 259, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/a3f6e51d-9789-4183-aed8-1c790f609a1f/targets/github%2Fsynthtool;config=default/tests;query=nodejs-firestore-session;failed=false).

Autosynth is still having trouble generating nodejs-firestore-session. :sob:

Here's the output from running `synth.py`:

```
s
.github/ISSUE_TEMPLATE/bug_report.md
.github/ISSUE_TEMPLATE/feature_request.md
.github/ISSUE_TEMPLATE/support_request.md
.github/PULL_REQUEST_TEMPLATE.md
.github/release-please.yml
.github/workflows/ci.yaml
.kokoro/.gitattributes
.kokoro/common.cfg
.kokoro/continuous/node10/common.cfg
.kokoro/continuous/node10/docs.cfg
.kokoro/continuous/node10/test.cfg
.kokoro/continuous/node12/common.cfg
.kokoro/continuous/node12/lint.cfg
.kokoro/continuous/node12/samples-test.cfg
.kokoro/continuous/node12/system-test.cfg
.kokoro/continuous/node12/test.cfg
.kokoro/docs.sh
.kokoro/lint.sh
.kokoro/populate-secrets.sh
.kokoro/presubmit/node10/common.cfg
.kokoro/presubmit/node12/common.cfg
.kokoro/presubmit/node12/samples-test.cfg
.kokoro/presubmit/node12/system-test.cfg
.kokoro/presubmit/node12/test.cfg
.kokoro/publish.sh
.kokoro/release/docs-devsite.cfg
.kokoro/release/docs-devsite.sh
.kokoro/release/docs.cfg
.kokoro/release/docs.sh
.kokoro/release/publish.cfg
.kokoro/samples-test.sh
.kokoro/system-test.sh
.kokoro/test.bat
.kokoro/test.sh
.kokoro/trampoline.sh
.kokoro/trampoline_v2.sh
.mocharc.js
.nycrc
.prettierignore
.prettierrc.js
.trampolinerc
CODE_OF_CONDUCT.md
CONTRIBUTING.md
LICENSE
README.md
api-extractor.json
renovate.json
samples/README.md
2020-11-14 02:54:33,474 synthtool [DEBUG] > Installing dependencies...
DEBUG:synthtool:Installing dependencies...
npm WARN deprecated ts-simple-ast@12.4.0: NOTICE: ts-simple-ast has been renamed to ts-morph and version reset to 1.0.0. Switch at your leisure...
npm WARN deprecated chokidar@2.1.8: Chokidar 2 will break on node v14+. Upgrade to chokidar 3 with 15x less dependencies.
npm WARN deprecated opn@6.0.0: The package has been renamed to `open`
npm WARN deprecated fsevents@1.2.13: fsevents 1 will break on node v14+ and could be using insecure binaries. Upgrade to fsevents 2.
npm WARN deprecated core-js@2.6.11: core-js@<3 is no longer maintained and not recommended for usage due to the number of issues. Please, upgrade your dependencies to the actual version of core-js@3.
npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated

> core-js@2.6.11 postinstall /home/kbuilder/.cache/synthtool/nodejs-firestore-session/node_modules/core-js
> node -e "try{require('./postinstall')}catch(e){}"

[96mThank you for using core-js ([94m https://github.com/zloirock/core-js [96m) for polyfilling JavaScript standard library![0m

[96mThe project needs your help! Please consider supporting of core-js on Open Collective or Patreon: [0m
[96m>[94m https://opencollective.com/core-js [0m
[96m>[94m https://www.patreon.com/zloirock [0m

[96mAlso, the author of core-js ([94m https://github.com/zloirock [96m) is looking for a good job -)[0m


> protobufjs@6.10.1 postinstall /home/kbuilder/.cache/synthtool/nodejs-firestore-session/node_modules/protobufjs
> node scripts/postinstall


> @compodoc/compodoc@1.1.11 postinstall /home/kbuilder/.cache/synthtool/nodejs-firestore-session/node_modules/@compodoc/compodoc
> opencollective-postinstall || exit 0

[96m[1mThank you for using @compodoc/compodoc![96m[1m
[0m[96mIf you rely on this package, please consider supporting our open collective:[22m[39m
> [94mhttps://opencollective.com/compodoc/donate[0m


> @google-cloud/connect-firestore@2.0.0 prepare /home/kbuilder/.cache/synthtool/nodejs-firestore-session
> npm run compile


> @google-cloud/connect-firestore@2.0.0 precompile /home/kbuilder/.cache/synthtool/nodejs-firestore-session
> gts clean

version: 14
Removing build ...

> @google-cloud/connect-firestore@2.0.0 compile /home/kbuilder/.cache/synthtool/nodejs-firestore-session
> tsc -p .

[96msrc/index.ts[0m:[93m27[0m:[93m11[0m - [91merror[0m[90m TS2559: [0mType 'StoreOptions' has no properties in common with type 'EventEmitterOptions'.

[7m27[0m     super(options || {});
[7m  [0m [91m          ~~~~~~~~~~~~~[0m

[96msrc/index.ts[0m:[93m37[0m:[93m54[0m - [91merror[0m[90m TS2694: [0mNamespace 'global.Express' has no exported member 'SessionData'.

[7m37[0m     callback: (err?: Error | null, session?: Express.SessionData) => void
[7m  [0m [91m                                                     ~~~~~~~~~~~[0m

[96msrc/index.ts[0m:[93m59[0m:[93m22[0m - [91merror[0m[90m TS2694: [0mNamespace 'global.Express' has no exported member 'SessionData'.

[7m59[0m     session: Express.SessionData,
[7m  [0m [91m                     ~~~~~~~~~~~[0m

[96msystem-test/session.ts[0m:[93m34[0m:[93m57[0m - [91merror[0m[90m TS2694: [0mNamespace 'global.Express' has no exported member 'SessionData'.

[7m34[0m     const sessionData = ({foo: 'bar'} as {}) as Express.SessionData;
[7m  [0m [91m                                                        ~~~~~~~~~~~[0m

[96mtest/test.ts[0m:[93m99[0m:[93m42[0m - [91merror[0m[90m TS2694: [0mNamespace 'global.Express' has no exported member 'SessionData'.

[7m99[0m     store.set(expectedSid, {} as Express.SessionData, assert.ifError);
[7m  [0m [91m                                         ~~~~~~~~~~~[0m


Found 5 errors.

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @google-cloud/connect-firestore@2.0.0 compile: `tsc -p .`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @google-cloud/connect-firestore@2.0.0 compile script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/kbuilder/.npm/_logs/2020-11-14T10_55_01_218Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @google-cloud/connect-firestore@2.0.0 prepare: `npm run compile`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @google-cloud/connect-firestore@2.0.0 prepare script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/kbuilder/.npm/_logs/2020-11-14T10_55_01_300Z-debug.log
2020-11-14 02:55:01,341 synthtool [ERROR] > Failed executing npm install:

None
ERROR:synthtool:Failed executing npm install:

None
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/nodejs-firestore-session/synth.py", line 15, in <module>
    node.install()
  File "/tmpfs/src/github/synthtool/synthtool/languages/node.py", line 167, in install
    shell.run(["npm", "install"], hide_output=hide_output)
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 39, in run
    raise exc
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 33, in run
    encoding="utf-8",
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['npm', 'install']' returned non-zero exit status 1.
2020-11-14 02:55:01,523 autosynth [ERROR] > Synthesis failed
2020-11-14 02:55:01,523 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at b623c57 build(node): add KOKORO_BUILD_ARTIFACTS_SUBDIR to env (#147)
2020-11-14 02:55:01,531 autosynth [DEBUG] > Running: git checkout autosynth
Switched to branch 'autosynth'
2020-11-14 02:55:01,537 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Removing node_modules/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 354, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 189, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 334, in _inner_main
    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 65, in synthesize_loop
    has_changes = toolbox.synthesize_version_in_new_branch(synthesizer, youngest)
  File "/tmpfs/src/github/synthtool/autosynth/synth_toolbox.py", line 259, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/f3c67c75-f169-4302-992f-c7bc582020d8/targets/github%2Fsynthtool;config=default/tests;query=nodejs-firestore-session;failed=false).

Autosynth is still having trouble generating nodejs-firestore-session. :sob:

Here's the output from running `synth.py`:

```
htool --metadata synth.metadata synth.py --
2020-11-15 02:51:38,815 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/nodejs-firestore-session/synth.py.
On branch autosynth-9
nothing to commit, working tree clean
2020-11-15 02:51:38,942 synthtool [DEBUG] > Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
DEBUG:synthtool:Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
.eslintignore
.eslintrc.json
.gitattributes
.github/ISSUE_TEMPLATE/bug_report.md
.github/ISSUE_TEMPLATE/feature_request.md
.github/ISSUE_TEMPLATE/support_request.md
.github/PULL_REQUEST_TEMPLATE.md
.github/release-please.yml
.github/workflows/ci.yaml
.kokoro/.gitattributes
.kokoro/common.cfg
.kokoro/continuous/node10/common.cfg
.kokoro/continuous/node10/docs.cfg
.kokoro/continuous/node10/test.cfg
.kokoro/continuous/node12/common.cfg
.kokoro/continuous/node12/lint.cfg
.kokoro/continuous/node12/samples-test.cfg
.kokoro/continuous/node12/system-test.cfg
.kokoro/continuous/node12/test.cfg
.kokoro/docs.sh
.kokoro/lint.sh
.kokoro/populate-secrets.sh
.kokoro/presubmit/node10/common.cfg
.kokoro/presubmit/node12/common.cfg
.kokoro/presubmit/node12/samples-test.cfg
.kokoro/presubmit/node12/system-test.cfg
.kokoro/presubmit/node12/test.cfg
.kokoro/publish.sh
.kokoro/release/docs-devsite.cfg
.kokoro/release/docs-devsite.sh
.kokoro/release/docs.cfg
.kokoro/release/docs.sh
.kokoro/release/publish.cfg
.kokoro/samples-test.sh
.kokoro/system-test.sh
.kokoro/test.bat
.kokoro/test.sh
.kokoro/trampoline.sh
.kokoro/trampoline_v2.sh
.mocharc.js
.nycrc
.prettierignore
.prettierrc.js
.trampolinerc
CODE_OF_CONDUCT.md
CONTRIBUTING.md
LICENSE
README.md
api-extractor.json
renovate.json
samples/README.md
2020-11-15 02:51:39,109 synthtool [DEBUG] > Installing dependencies...
DEBUG:synthtool:Installing dependencies...
npm WARN deprecated ts-simple-ast@12.4.0: NOTICE: ts-simple-ast has been renamed to ts-morph and version reset to 1.0.0. Switch at your leisure...
npm WARN deprecated chokidar@2.1.8: Chokidar 2 will break on node v14+. Upgrade to chokidar 3 with 15x less dependencies.
npm WARN deprecated opn@6.0.0: The package has been renamed to `open`
npm WARN deprecated fsevents@1.2.13: fsevents 1 will break on node v14+ and could be using insecure binaries. Upgrade to fsevents 2.
npm WARN deprecated core-js@2.6.11: core-js@<3 is no longer maintained and not recommended for usage due to the number of issues. Please, upgrade your dependencies to the actual version of core-js@3.
npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated

> core-js@2.6.11 postinstall /home/kbuilder/.cache/synthtool/nodejs-firestore-session/node_modules/core-js
> node -e "try{require('./postinstall')}catch(e){}"


> protobufjs@6.10.1 postinstall /home/kbuilder/.cache/synthtool/nodejs-firestore-session/node_modules/protobufjs
> node scripts/postinstall


> @compodoc/compodoc@1.1.11 postinstall /home/kbuilder/.cache/synthtool/nodejs-firestore-session/node_modules/@compodoc/compodoc
> opencollective-postinstall || exit 0

[96m[1mThank you for using @compodoc/compodoc![96m[1m
[0m[96mIf you rely on this package, please consider supporting our open collective:[22m[39m
> [94mhttps://opencollective.com/compodoc/donate[0m


> @google-cloud/connect-firestore@2.0.0 prepare /home/kbuilder/.cache/synthtool/nodejs-firestore-session
> npm run compile


> @google-cloud/connect-firestore@2.0.0 precompile /home/kbuilder/.cache/synthtool/nodejs-firestore-session
> gts clean

version: 14
Removing build ...

> @google-cloud/connect-firestore@2.0.0 compile /home/kbuilder/.cache/synthtool/nodejs-firestore-session
> tsc -p .

[96msrc/index.ts[0m:[93m27[0m:[93m11[0m - [91merror[0m[90m TS2559: [0mType 'StoreOptions' has no properties in common with type 'EventEmitterOptions'.

[7m27[0m     super(options || {});
[7m  [0m [91m          ~~~~~~~~~~~~~[0m

[96msrc/index.ts[0m:[93m37[0m:[93m54[0m - [91merror[0m[90m TS2694: [0mNamespace 'global.Express' has no exported member 'SessionData'.

[7m37[0m     callback: (err?: Error | null, session?: Express.SessionData) => void
[7m  [0m [91m                                                     ~~~~~~~~~~~[0m

[96msrc/index.ts[0m:[93m59[0m:[93m22[0m - [91merror[0m[90m TS2694: [0mNamespace 'global.Express' has no exported member 'SessionData'.

[7m59[0m     session: Express.SessionData,
[7m  [0m [91m                     ~~~~~~~~~~~[0m

[96msystem-test/session.ts[0m:[93m34[0m:[93m57[0m - [91merror[0m[90m TS2694: [0mNamespace 'global.Express' has no exported member 'SessionData'.

[7m34[0m     const sessionData = ({foo: 'bar'} as {}) as Express.SessionData;
[7m  [0m [91m                                                        ~~~~~~~~~~~[0m

[96mtest/test.ts[0m:[93m99[0m:[93m42[0m - [91merror[0m[90m TS2694: [0mNamespace 'global.Express' has no exported member 'SessionData'.

[7m99[0m     store.set(expectedSid, {} as Express.SessionData, assert.ifError);
[7m  [0m [91m                                         ~~~~~~~~~~~[0m


Found 5 errors.

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @google-cloud/connect-firestore@2.0.0 compile: `tsc -p .`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @google-cloud/connect-firestore@2.0.0 compile script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/kbuilder/.npm/_logs/2020-11-15T10_52_01_849Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @google-cloud/connect-firestore@2.0.0 prepare: `npm run compile`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @google-cloud/connect-firestore@2.0.0 prepare script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/kbuilder/.npm/_logs/2020-11-15T10_52_01_913Z-debug.log
2020-11-15 02:52:01,966 synthtool [ERROR] > Failed executing npm install:

None
ERROR:synthtool:Failed executing npm install:

None
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/nodejs-firestore-session/synth.py", line 15, in <module>
    node.install()
  File "/tmpfs/src/github/synthtool/synthtool/languages/node.py", line 167, in install
    shell.run(["npm", "install"], hide_output=hide_output)
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 39, in run
    raise exc
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 33, in run
    encoding="utf-8",
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['npm', 'install']' returned non-zero exit status 1.
2020-11-15 02:52:02,105 autosynth [ERROR] > Synthesis failed
2020-11-15 02:52:02,105 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at b623c57 build(node): add KOKORO_BUILD_ARTIFACTS_SUBDIR to env (#147)
2020-11-15 02:52:02,113 autosynth [DEBUG] > Running: git checkout autosynth
Switched to branch 'autosynth'
2020-11-15 02:52:02,118 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Removing node_modules/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 354, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 189, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 334, in _inner_main
    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 65, in synthesize_loop
    has_changes = toolbox.synthesize_version_in_new_branch(synthesizer, youngest)
  File "/tmpfs/src/github/synthtool/autosynth/synth_toolbox.py", line 259, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/b6c13fe2-2ad6-4bef-875d-0d581117b03d/targets/github%2Fsynthtool;config=default/tests;query=nodejs-firestore-session;failed=false).

Autosynth is still having trouble generating nodejs-firestore-session. :sob:

Here's the output from running `synth.py`:

```
s
.github/ISSUE_TEMPLATE/bug_report.md
.github/ISSUE_TEMPLATE/feature_request.md
.github/ISSUE_TEMPLATE/support_request.md
.github/PULL_REQUEST_TEMPLATE.md
.github/release-please.yml
.github/workflows/ci.yaml
.kokoro/.gitattributes
.kokoro/common.cfg
.kokoro/continuous/node10/common.cfg
.kokoro/continuous/node10/docs.cfg
.kokoro/continuous/node10/test.cfg
.kokoro/continuous/node12/common.cfg
.kokoro/continuous/node12/lint.cfg
.kokoro/continuous/node12/samples-test.cfg
.kokoro/continuous/node12/system-test.cfg
.kokoro/continuous/node12/test.cfg
.kokoro/docs.sh
.kokoro/lint.sh
.kokoro/populate-secrets.sh
.kokoro/presubmit/node10/common.cfg
.kokoro/presubmit/node12/common.cfg
.kokoro/presubmit/node12/samples-test.cfg
.kokoro/presubmit/node12/system-test.cfg
.kokoro/presubmit/node12/test.cfg
.kokoro/publish.sh
.kokoro/release/docs-devsite.cfg
.kokoro/release/docs-devsite.sh
.kokoro/release/docs.cfg
.kokoro/release/docs.sh
.kokoro/release/publish.cfg
.kokoro/samples-test.sh
.kokoro/system-test.sh
.kokoro/test.bat
.kokoro/test.sh
.kokoro/trampoline.sh
.kokoro/trampoline_v2.sh
.mocharc.js
.nycrc
.prettierignore
.prettierrc.js
.trampolinerc
CODE_OF_CONDUCT.md
CONTRIBUTING.md
LICENSE
README.md
api-extractor.json
renovate.json
samples/README.md
2020-11-16 02:50:19,166 synthtool [DEBUG] > Installing dependencies...
DEBUG:synthtool:Installing dependencies...
npm WARN deprecated ts-simple-ast@12.4.0: NOTICE: ts-simple-ast has been renamed to ts-morph and version reset to 1.0.0. Switch at your leisure...
npm WARN deprecated chokidar@2.1.8: Chokidar 2 will break on node v14+. Upgrade to chokidar 3 with 15x less dependencies.
npm WARN deprecated opn@6.0.0: The package has been renamed to `open`
npm WARN deprecated fsevents@1.2.13: fsevents 1 will break on node v14+ and could be using insecure binaries. Upgrade to fsevents 2.
npm WARN deprecated core-js@2.6.11: core-js@<3 is no longer maintained and not recommended for usage due to the number of issues. Please, upgrade your dependencies to the actual version of core-js@3.
npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated

> core-js@2.6.11 postinstall /home/kbuilder/.cache/synthtool/nodejs-firestore-session/node_modules/core-js
> node -e "try{require('./postinstall')}catch(e){}"

[96mThank you for using core-js ([94m https://github.com/zloirock/core-js [96m) for polyfilling JavaScript standard library![0m

[96mThe project needs your help! Please consider supporting of core-js on Open Collective or Patreon: [0m
[96m>[94m https://opencollective.com/core-js [0m
[96m>[94m https://www.patreon.com/zloirock [0m

[96mAlso, the author of core-js ([94m https://github.com/zloirock [96m) is looking for a good job -)[0m


> protobufjs@6.10.1 postinstall /home/kbuilder/.cache/synthtool/nodejs-firestore-session/node_modules/protobufjs
> node scripts/postinstall


> @compodoc/compodoc@1.1.11 postinstall /home/kbuilder/.cache/synthtool/nodejs-firestore-session/node_modules/@compodoc/compodoc
> opencollective-postinstall || exit 0

[96m[1mThank you for using @compodoc/compodoc![96m[1m
[0m[96mIf you rely on this package, please consider supporting our open collective:[22m[39m
> [94mhttps://opencollective.com/compodoc/donate[0m


> @google-cloud/connect-firestore@2.0.0 prepare /home/kbuilder/.cache/synthtool/nodejs-firestore-session
> npm run compile


> @google-cloud/connect-firestore@2.0.0 precompile /home/kbuilder/.cache/synthtool/nodejs-firestore-session
> gts clean

version: 14
Removing build ...

> @google-cloud/connect-firestore@2.0.0 compile /home/kbuilder/.cache/synthtool/nodejs-firestore-session
> tsc -p .

[96msrc/index.ts[0m:[93m27[0m:[93m11[0m - [91merror[0m[90m TS2559: [0mType 'StoreOptions' has no properties in common with type 'EventEmitterOptions'.

[7m27[0m     super(options || {});
[7m  [0m [91m          ~~~~~~~~~~~~~[0m

[96msrc/index.ts[0m:[93m37[0m:[93m54[0m - [91merror[0m[90m TS2694: [0mNamespace 'global.Express' has no exported member 'SessionData'.

[7m37[0m     callback: (err?: Error | null, session?: Express.SessionData) => void
[7m  [0m [91m                                                     ~~~~~~~~~~~[0m

[96msrc/index.ts[0m:[93m59[0m:[93m22[0m - [91merror[0m[90m TS2694: [0mNamespace 'global.Express' has no exported member 'SessionData'.

[7m59[0m     session: Express.SessionData,
[7m  [0m [91m                     ~~~~~~~~~~~[0m

[96msystem-test/session.ts[0m:[93m34[0m:[93m57[0m - [91merror[0m[90m TS2694: [0mNamespace 'global.Express' has no exported member 'SessionData'.

[7m34[0m     const sessionData = ({foo: 'bar'} as {}) as Express.SessionData;
[7m  [0m [91m                                                        ~~~~~~~~~~~[0m

[96mtest/test.ts[0m:[93m99[0m:[93m42[0m - [91merror[0m[90m TS2694: [0mNamespace 'global.Express' has no exported member 'SessionData'.

[7m99[0m     store.set(expectedSid, {} as Express.SessionData, assert.ifError);
[7m  [0m [91m                                         ~~~~~~~~~~~[0m


Found 5 errors.

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @google-cloud/connect-firestore@2.0.0 compile: `tsc -p .`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @google-cloud/connect-firestore@2.0.0 compile script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/kbuilder/.npm/_logs/2020-11-16T10_50_43_809Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @google-cloud/connect-firestore@2.0.0 prepare: `npm run compile`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @google-cloud/connect-firestore@2.0.0 prepare script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/kbuilder/.npm/_logs/2020-11-16T10_50_43_892Z-debug.log
2020-11-16 02:50:43,928 synthtool [ERROR] > Failed executing npm install:

None
ERROR:synthtool:Failed executing npm install:

None
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/nodejs-firestore-session/synth.py", line 15, in <module>
    node.install()
  File "/tmpfs/src/github/synthtool/synthtool/languages/node.py", line 167, in install
    shell.run(["npm", "install"], hide_output=hide_output)
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 39, in run
    raise exc
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 33, in run
    encoding="utf-8",
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['npm', 'install']' returned non-zero exit status 1.
2020-11-16 02:50:44,065 autosynth [ERROR] > Synthesis failed
2020-11-16 02:50:44,065 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at b623c57 build(node): add KOKORO_BUILD_ARTIFACTS_SUBDIR to env (#147)
2020-11-16 02:50:44,074 autosynth [DEBUG] > Running: git checkout autosynth
Switched to branch 'autosynth'
2020-11-16 02:50:44,079 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Removing node_modules/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 354, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 189, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 334, in _inner_main
    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 65, in synthesize_loop
    has_changes = toolbox.synthesize_version_in_new_branch(synthesizer, youngest)
  File "/tmpfs/src/github/synthtool/autosynth/synth_toolbox.py", line 259, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/df415faf-fd58-443d-9921-b0652b681b80/targets/github%2Fsynthtool;config=default/tests;query=nodejs-firestore-session;failed=false).

Autosynth is still having trouble generating nodejs-firestore-session. :sob:

Here's the output from running `synth.py`:

```
htool --metadata synth.metadata synth.py --
2020-11-17 02:44:58,173 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/nodejs-firestore-session/synth.py.
On branch autosynth-9
nothing to commit, working tree clean
2020-11-17 02:44:58,304 synthtool [DEBUG] > Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
DEBUG:synthtool:Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
.eslintignore
.eslintrc.json
.gitattributes
.github/ISSUE_TEMPLATE/bug_report.md
.github/ISSUE_TEMPLATE/feature_request.md
.github/ISSUE_TEMPLATE/support_request.md
.github/PULL_REQUEST_TEMPLATE.md
.github/release-please.yml
.github/workflows/ci.yaml
.kokoro/.gitattributes
.kokoro/common.cfg
.kokoro/continuous/node10/common.cfg
.kokoro/continuous/node10/docs.cfg
.kokoro/continuous/node10/test.cfg
.kokoro/continuous/node12/common.cfg
.kokoro/continuous/node12/lint.cfg
.kokoro/continuous/node12/samples-test.cfg
.kokoro/continuous/node12/system-test.cfg
.kokoro/continuous/node12/test.cfg
.kokoro/docs.sh
.kokoro/lint.sh
.kokoro/populate-secrets.sh
.kokoro/presubmit/node10/common.cfg
.kokoro/presubmit/node12/common.cfg
.kokoro/presubmit/node12/samples-test.cfg
.kokoro/presubmit/node12/system-test.cfg
.kokoro/presubmit/node12/test.cfg
.kokoro/publish.sh
.kokoro/release/docs-devsite.cfg
.kokoro/release/docs-devsite.sh
.kokoro/release/docs.cfg
.kokoro/release/docs.sh
.kokoro/release/publish.cfg
.kokoro/samples-test.sh
.kokoro/system-test.sh
.kokoro/test.bat
.kokoro/test.sh
.kokoro/trampoline.sh
.kokoro/trampoline_v2.sh
.mocharc.js
.nycrc
.prettierignore
.prettierrc.js
.trampolinerc
CODE_OF_CONDUCT.md
CONTRIBUTING.md
LICENSE
README.md
api-extractor.json
renovate.json
samples/README.md
2020-11-17 02:44:58,481 synthtool [DEBUG] > Installing dependencies...
DEBUG:synthtool:Installing dependencies...
npm WARN deprecated ts-simple-ast@12.4.0: NOTICE: ts-simple-ast has been renamed to ts-morph and version reset to 1.0.0. Switch at your leisure...
npm WARN deprecated chokidar@2.1.8: Chokidar 2 will break on node v14+. Upgrade to chokidar 3 with 15x less dependencies.
npm WARN deprecated opn@6.0.0: The package has been renamed to `open`
npm WARN deprecated fsevents@1.2.13: fsevents 1 will break on node v14+ and could be using insecure binaries. Upgrade to fsevents 2.
npm WARN deprecated core-js@2.6.11: core-js@<3 is no longer maintained and not recommended for usage due to the number of issues. Please, upgrade your dependencies to the actual version of core-js@3.
npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated

> core-js@2.6.11 postinstall /home/kbuilder/.cache/synthtool/nodejs-firestore-session/node_modules/core-js
> node -e "try{require('./postinstall')}catch(e){}"


> protobufjs@6.10.2 postinstall /home/kbuilder/.cache/synthtool/nodejs-firestore-session/node_modules/protobufjs
> node scripts/postinstall


> @compodoc/compodoc@1.1.11 postinstall /home/kbuilder/.cache/synthtool/nodejs-firestore-session/node_modules/@compodoc/compodoc
> opencollective-postinstall || exit 0

[96m[1mThank you for using @compodoc/compodoc![96m[1m
[0m[96mIf you rely on this package, please consider supporting our open collective:[22m[39m
> [94mhttps://opencollective.com/compodoc/donate[0m


> @google-cloud/connect-firestore@2.0.0 prepare /home/kbuilder/.cache/synthtool/nodejs-firestore-session
> npm run compile


> @google-cloud/connect-firestore@2.0.0 precompile /home/kbuilder/.cache/synthtool/nodejs-firestore-session
> gts clean

version: 14
Removing build ...

> @google-cloud/connect-firestore@2.0.0 compile /home/kbuilder/.cache/synthtool/nodejs-firestore-session
> tsc -p .

[96msrc/index.ts[0m:[93m27[0m:[93m11[0m - [91merror[0m[90m TS2559: [0mType 'StoreOptions' has no properties in common with type 'EventEmitterOptions'.

[7m27[0m     super(options || {});
[7m  [0m [91m          ~~~~~~~~~~~~~[0m

[96msrc/index.ts[0m:[93m37[0m:[93m54[0m - [91merror[0m[90m TS2694: [0mNamespace 'global.Express' has no exported member 'SessionData'.

[7m37[0m     callback: (err?: Error | null, session?: Express.SessionData) => void
[7m  [0m [91m                                                     ~~~~~~~~~~~[0m

[96msrc/index.ts[0m:[93m59[0m:[93m22[0m - [91merror[0m[90m TS2694: [0mNamespace 'global.Express' has no exported member 'SessionData'.

[7m59[0m     session: Express.SessionData,
[7m  [0m [91m                     ~~~~~~~~~~~[0m

[96msystem-test/session.ts[0m:[93m34[0m:[93m57[0m - [91merror[0m[90m TS2694: [0mNamespace 'global.Express' has no exported member 'SessionData'.

[7m34[0m     const sessionData = ({foo: 'bar'} as {}) as Express.SessionData;
[7m  [0m [91m                                                        ~~~~~~~~~~~[0m

[96mtest/test.ts[0m:[93m99[0m:[93m42[0m - [91merror[0m[90m TS2694: [0mNamespace 'global.Express' has no exported member 'SessionData'.

[7m99[0m     store.set(expectedSid, {} as Express.SessionData, assert.ifError);
[7m  [0m [91m                                         ~~~~~~~~~~~[0m


Found 5 errors.

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @google-cloud/connect-firestore@2.0.0 compile: `tsc -p .`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @google-cloud/connect-firestore@2.0.0 compile script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/kbuilder/.npm/_logs/2020-11-17T10_45_23_261Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @google-cloud/connect-firestore@2.0.0 prepare: `npm run compile`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @google-cloud/connect-firestore@2.0.0 prepare script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/kbuilder/.npm/_logs/2020-11-17T10_45_23_349Z-debug.log
2020-11-17 02:45:23,388 synthtool [ERROR] > Failed executing npm install:

None
ERROR:synthtool:Failed executing npm install:

None
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/nodejs-firestore-session/synth.py", line 15, in <module>
    node.install()
  File "/tmpfs/src/github/synthtool/synthtool/languages/node.py", line 167, in install
    shell.run(["npm", "install"], hide_output=hide_output)
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 39, in run
    raise exc
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 33, in run
    encoding="utf-8",
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['npm', 'install']' returned non-zero exit status 1.
2020-11-17 02:45:23,755 autosynth [ERROR] > Synthesis failed
2020-11-17 02:45:23,755 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at b623c57 build(node): add KOKORO_BUILD_ARTIFACTS_SUBDIR to env (#147)
2020-11-17 02:45:23,762 autosynth [DEBUG] > Running: git checkout autosynth
Switched to branch 'autosynth'
2020-11-17 02:45:23,767 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Removing node_modules/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 354, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 189, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 334, in _inner_main
    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 65, in synthesize_loop
    has_changes = toolbox.synthesize_version_in_new_branch(synthesizer, youngest)
  File "/tmpfs/src/github/synthtool/autosynth/synth_toolbox.py", line 259, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/56efccb8-6d0a-4d51-9b5c-a1f1439cf086/targets/github%2Fsynthtool;config=default/tests;query=nodejs-firestore-session;failed=false).

Autosynth is still having trouble generating nodejs-firestore-session. :sob:

Here's the output from running `synth.py`:

```
tool --metadata synth.metadata synth.py --
2020-11-18 02:41:46,012 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/nodejs-firestore-session/synth.py.
On branch autosynth-10
nothing to commit, working tree clean
2020-11-18 02:41:46,132 synthtool [DEBUG] > Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
DEBUG:synthtool:Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
.eslintignore
.eslintrc.json
.gitattributes
.github/ISSUE_TEMPLATE/bug_report.md
.github/ISSUE_TEMPLATE/feature_request.md
.github/ISSUE_TEMPLATE/support_request.md
.github/PULL_REQUEST_TEMPLATE.md
.github/release-please.yml
.github/workflows/ci.yaml
.kokoro/.gitattributes
.kokoro/common.cfg
.kokoro/continuous/node10/common.cfg
.kokoro/continuous/node10/docs.cfg
.kokoro/continuous/node10/test.cfg
.kokoro/continuous/node12/common.cfg
.kokoro/continuous/node12/lint.cfg
.kokoro/continuous/node12/samples-test.cfg
.kokoro/continuous/node12/system-test.cfg
.kokoro/continuous/node12/test.cfg
.kokoro/docs.sh
.kokoro/lint.sh
.kokoro/populate-secrets.sh
.kokoro/presubmit/node10/common.cfg
.kokoro/presubmit/node12/common.cfg
.kokoro/presubmit/node12/samples-test.cfg
.kokoro/presubmit/node12/system-test.cfg
.kokoro/presubmit/node12/test.cfg
.kokoro/publish.sh
.kokoro/release/docs-devsite.cfg
.kokoro/release/docs-devsite.sh
.kokoro/release/docs.cfg
.kokoro/release/docs.sh
.kokoro/release/publish.cfg
.kokoro/samples-test.sh
.kokoro/system-test.sh
.kokoro/test.bat
.kokoro/test.sh
.kokoro/trampoline.sh
.kokoro/trampoline_v2.sh
.mocharc.js
.nycrc
.prettierignore
.prettierrc.js
.trampolinerc
CODE_OF_CONDUCT.md
CONTRIBUTING.md
LICENSE
README.md
api-extractor.json
renovate.json
samples/README.md
2020-11-18 02:41:46,293 synthtool [DEBUG] > Installing dependencies...
DEBUG:synthtool:Installing dependencies...
npm WARN deprecated ts-simple-ast@12.4.0: NOTICE: ts-simple-ast has been renamed to ts-morph and version reset to 1.0.0. Switch at your leisure...
npm WARN deprecated chokidar@2.1.8: Chokidar 2 will break on node v14+. Upgrade to chokidar 3 with 15x less dependencies.
npm WARN deprecated opn@6.0.0: The package has been renamed to `open`
npm WARN deprecated fsevents@1.2.13: fsevents 1 will break on node v14+ and could be using insecure binaries. Upgrade to fsevents 2.
npm WARN deprecated core-js@2.6.11: core-js@<3 is no longer maintained and not recommended for usage due to the number of issues. Please, upgrade your dependencies to the actual version of core-js@3.
npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated

> core-js@2.6.11 postinstall /home/kbuilder/.cache/synthtool/nodejs-firestore-session/node_modules/core-js
> node -e "try{require('./postinstall')}catch(e){}"


> protobufjs@6.10.2 postinstall /home/kbuilder/.cache/synthtool/nodejs-firestore-session/node_modules/protobufjs
> node scripts/postinstall


> @compodoc/compodoc@1.1.11 postinstall /home/kbuilder/.cache/synthtool/nodejs-firestore-session/node_modules/@compodoc/compodoc
> opencollective-postinstall || exit 0

[96m[1mThank you for using @compodoc/compodoc![96m[1m
[0m[96mIf you rely on this package, please consider supporting our open collective:[22m[39m
> [94mhttps://opencollective.com/compodoc/donate[0m


> @google-cloud/connect-firestore@2.0.0 prepare /home/kbuilder/.cache/synthtool/nodejs-firestore-session
> npm run compile


> @google-cloud/connect-firestore@2.0.0 precompile /home/kbuilder/.cache/synthtool/nodejs-firestore-session
> gts clean

version: 14
Removing build ...

> @google-cloud/connect-firestore@2.0.0 compile /home/kbuilder/.cache/synthtool/nodejs-firestore-session
> tsc -p .

[96msrc/index.ts[0m:[93m27[0m:[93m11[0m - [91merror[0m[90m TS2559: [0mType 'StoreOptions' has no properties in common with type 'EventEmitterOptions'.

[7m27[0m     super(options || {});
[7m  [0m [91m          ~~~~~~~~~~~~~[0m

[96msrc/index.ts[0m:[93m37[0m:[93m54[0m - [91merror[0m[90m TS2694: [0mNamespace 'global.Express' has no exported member 'SessionData'.

[7m37[0m     callback: (err?: Error | null, session?: Express.SessionData) => void
[7m  [0m [91m                                                     ~~~~~~~~~~~[0m

[96msrc/index.ts[0m:[93m59[0m:[93m22[0m - [91merror[0m[90m TS2694: [0mNamespace 'global.Express' has no exported member 'SessionData'.

[7m59[0m     session: Express.SessionData,
[7m  [0m [91m                     ~~~~~~~~~~~[0m

[96msystem-test/session.ts[0m:[93m34[0m:[93m57[0m - [91merror[0m[90m TS2694: [0mNamespace 'global.Express' has no exported member 'SessionData'.

[7m34[0m     const sessionData = ({foo: 'bar'} as {}) as Express.SessionData;
[7m  [0m [91m                                                        ~~~~~~~~~~~[0m

[96mtest/test.ts[0m:[93m99[0m:[93m42[0m - [91merror[0m[90m TS2694: [0mNamespace 'global.Express' has no exported member 'SessionData'.

[7m99[0m     store.set(expectedSid, {} as Express.SessionData, assert.ifError);
[7m  [0m [91m                                         ~~~~~~~~~~~[0m


Found 5 errors.

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @google-cloud/connect-firestore@2.0.0 compile: `tsc -p .`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @google-cloud/connect-firestore@2.0.0 compile script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/kbuilder/.npm/_logs/2020-11-18T10_42_09_560Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @google-cloud/connect-firestore@2.0.0 prepare: `npm run compile`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @google-cloud/connect-firestore@2.0.0 prepare script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/kbuilder/.npm/_logs/2020-11-18T10_42_09_620Z-debug.log
2020-11-18 02:42:09,673 synthtool [ERROR] > Failed executing npm install:

None
ERROR:synthtool:Failed executing npm install:

None
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/nodejs-firestore-session/synth.py", line 15, in <module>
    node.install()
  File "/tmpfs/src/github/synthtool/synthtool/languages/node.py", line 167, in install
    shell.run(["npm", "install"], hide_output=hide_output)
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 39, in run
    raise exc
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 33, in run
    encoding="utf-8",
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['npm', 'install']' returned non-zero exit status 1.
2020-11-18 02:42:09,801 autosynth [ERROR] > Synthesis failed
2020-11-18 02:42:09,801 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at b623c57 build(node): add KOKORO_BUILD_ARTIFACTS_SUBDIR to env (#147)
2020-11-18 02:42:09,808 autosynth [DEBUG] > Running: git checkout autosynth
Switched to branch 'autosynth'
2020-11-18 02:42:09,812 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Removing node_modules/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 354, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 189, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 334, in _inner_main
    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 65, in synthesize_loop
    has_changes = toolbox.synthesize_version_in_new_branch(synthesizer, youngest)
  File "/tmpfs/src/github/synthtool/autosynth/synth_toolbox.py", line 259, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/a16ab8f3-8935-483a-a45b-6d0182d4e70f/targets/github%2Fsynthtool;config=default/tests;query=nodejs-firestore-session;failed=false).

Autosynth is still having trouble generating nodejs-firestore-session. :sob:

Here's the output from running `synth.py`:

```
s
.github/ISSUE_TEMPLATE/bug_report.md
.github/ISSUE_TEMPLATE/feature_request.md
.github/ISSUE_TEMPLATE/support_request.md
.github/PULL_REQUEST_TEMPLATE.md
.github/release-please.yml
.github/workflows/ci.yaml
.kokoro/.gitattributes
.kokoro/common.cfg
.kokoro/continuous/node10/common.cfg
.kokoro/continuous/node10/docs.cfg
.kokoro/continuous/node10/test.cfg
.kokoro/continuous/node12/common.cfg
.kokoro/continuous/node12/lint.cfg
.kokoro/continuous/node12/samples-test.cfg
.kokoro/continuous/node12/system-test.cfg
.kokoro/continuous/node12/test.cfg
.kokoro/docs.sh
.kokoro/lint.sh
.kokoro/populate-secrets.sh
.kokoro/presubmit/node10/common.cfg
.kokoro/presubmit/node12/common.cfg
.kokoro/presubmit/node12/samples-test.cfg
.kokoro/presubmit/node12/system-test.cfg
.kokoro/presubmit/node12/test.cfg
.kokoro/publish.sh
.kokoro/release/docs-devsite.cfg
.kokoro/release/docs-devsite.sh
.kokoro/release/docs.cfg
.kokoro/release/docs.sh
.kokoro/release/publish.cfg
.kokoro/samples-test.sh
.kokoro/system-test.sh
.kokoro/test.bat
.kokoro/test.sh
.kokoro/trampoline.sh
.kokoro/trampoline_v2.sh
.mocharc.js
.nycrc
.prettierignore
.prettierrc.js
.trampolinerc
CODE_OF_CONDUCT.md
CONTRIBUTING.md
LICENSE
README.md
api-extractor.json
renovate.json
samples/README.md
2020-11-19 04:31:51,332 synthtool [DEBUG] > Installing dependencies...
DEBUG:synthtool:Installing dependencies...
npm WARN deprecated ts-simple-ast@12.4.0: NOTICE: ts-simple-ast has been renamed to ts-morph and version reset to 1.0.0. Switch at your leisure...
npm WARN deprecated chokidar@2.1.8: Chokidar 2 will break on node v14+. Upgrade to chokidar 3 with 15x less dependencies.
npm WARN deprecated opn@6.0.0: The package has been renamed to `open`
npm WARN deprecated fsevents@1.2.13: fsevents 1 will break on node v14+ and could be using insecure binaries. Upgrade to fsevents 2.
npm WARN deprecated core-js@2.6.11: core-js@<3 is no longer maintained and not recommended for usage due to the number of issues. Please, upgrade your dependencies to the actual version of core-js@3.
npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated

> core-js@2.6.11 postinstall /home/kbuilder/.cache/synthtool/nodejs-firestore-session/node_modules/core-js
> node -e "try{require('./postinstall')}catch(e){}"

[96mThank you for using core-js ([94m https://github.com/zloirock/core-js [96m) for polyfilling JavaScript standard library![0m

[96mThe project needs your help! Please consider supporting of core-js on Open Collective or Patreon: [0m
[96m>[94m https://opencollective.com/core-js [0m
[96m>[94m https://www.patreon.com/zloirock [0m

[96mAlso, the author of core-js ([94m https://github.com/zloirock [96m) is looking for a good job -)[0m


> protobufjs@6.10.2 postinstall /home/kbuilder/.cache/synthtool/nodejs-firestore-session/node_modules/protobufjs
> node scripts/postinstall


> @compodoc/compodoc@1.1.11 postinstall /home/kbuilder/.cache/synthtool/nodejs-firestore-session/node_modules/@compodoc/compodoc
> opencollective-postinstall || exit 0

[96m[1mThank you for using @compodoc/compodoc![96m[1m
[0m[96mIf you rely on this package, please consider supporting our open collective:[22m[39m
> [94mhttps://opencollective.com/compodoc/donate[0m


> @google-cloud/connect-firestore@2.0.0 prepare /home/kbuilder/.cache/synthtool/nodejs-firestore-session
> npm run compile


> @google-cloud/connect-firestore@2.0.0 precompile /home/kbuilder/.cache/synthtool/nodejs-firestore-session
> gts clean

version: 14
Removing build ...

> @google-cloud/connect-firestore@2.0.0 compile /home/kbuilder/.cache/synthtool/nodejs-firestore-session
> tsc -p .

[96msrc/index.ts[0m:[93m27[0m:[93m11[0m - [91merror[0m[90m TS2559: [0mType 'StoreOptions' has no properties in common with type 'EventEmitterOptions'.

[7m27[0m     super(options || {});
[7m  [0m [91m          ~~~~~~~~~~~~~[0m

[96msrc/index.ts[0m:[93m37[0m:[93m54[0m - [91merror[0m[90m TS2694: [0mNamespace 'global.Express' has no exported member 'SessionData'.

[7m37[0m     callback: (err?: Error | null, session?: Express.SessionData) => void
[7m  [0m [91m                                                     ~~~~~~~~~~~[0m

[96msrc/index.ts[0m:[93m59[0m:[93m22[0m - [91merror[0m[90m TS2694: [0mNamespace 'global.Express' has no exported member 'SessionData'.

[7m59[0m     session: Express.SessionData,
[7m  [0m [91m                     ~~~~~~~~~~~[0m

[96msystem-test/session.ts[0m:[93m34[0m:[93m57[0m - [91merror[0m[90m TS2694: [0mNamespace 'global.Express' has no exported member 'SessionData'.

[7m34[0m     const sessionData = ({foo: 'bar'} as {}) as Express.SessionData;
[7m  [0m [91m                                                        ~~~~~~~~~~~[0m

[96mtest/test.ts[0m:[93m99[0m:[93m42[0m - [91merror[0m[90m TS2694: [0mNamespace 'global.Express' has no exported member 'SessionData'.

[7m99[0m     store.set(expectedSid, {} as Express.SessionData, assert.ifError);
[7m  [0m [91m                                         ~~~~~~~~~~~[0m


Found 5 errors.

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @google-cloud/connect-firestore@2.0.0 compile: `tsc -p .`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @google-cloud/connect-firestore@2.0.0 compile script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/kbuilder/.npm/_logs/2020-11-19T12_32_20_132Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @google-cloud/connect-firestore@2.0.0 prepare: `npm run compile`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @google-cloud/connect-firestore@2.0.0 prepare script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/kbuilder/.npm/_logs/2020-11-19T12_32_20_223Z-debug.log
2020-11-19 04:32:20,264 synthtool [ERROR] > Failed executing npm install:

None
ERROR:synthtool:Failed executing npm install:

None
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/nodejs-firestore-session/synth.py", line 15, in <module>
    node.install()
  File "/tmpfs/src/github/synthtool/synthtool/languages/node.py", line 167, in install
    shell.run(["npm", "install"], hide_output=hide_output)
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 39, in run
    raise exc
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 33, in run
    encoding="utf-8",
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['npm', 'install']' returned non-zero exit status 1.
2020-11-19 04:32:20,527 autosynth [ERROR] > Synthesis failed
2020-11-19 04:32:20,527 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at b623c57 build(node): add KOKORO_BUILD_ARTIFACTS_SUBDIR to env (#147)
2020-11-19 04:32:20,536 autosynth [DEBUG] > Running: git checkout autosynth
Switched to branch 'autosynth'
2020-11-19 04:32:20,541 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Removing node_modules/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 354, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 189, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 334, in _inner_main
    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 65, in synthesize_loop
    has_changes = toolbox.synthesize_version_in_new_branch(synthesizer, youngest)
  File "/tmpfs/src/github/synthtool/autosynth/synth_toolbox.py", line 259, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/0374b1d3-a66c-4bac-957b-ebb38419e58f/targets/github%2Fsynthtool;config=default/tests;query=nodejs-firestore-session;failed=false).

Autosynth is still having trouble generating nodejs-firestore-session. :sob:

Here's the output from running `synth.py`:

```
s
.github/ISSUE_TEMPLATE/bug_report.md
.github/ISSUE_TEMPLATE/feature_request.md
.github/ISSUE_TEMPLATE/support_request.md
.github/PULL_REQUEST_TEMPLATE.md
.github/release-please.yml
.github/workflows/ci.yaml
.kokoro/.gitattributes
.kokoro/common.cfg
.kokoro/continuous/node10/common.cfg
.kokoro/continuous/node10/docs.cfg
.kokoro/continuous/node10/test.cfg
.kokoro/continuous/node12/common.cfg
.kokoro/continuous/node12/lint.cfg
.kokoro/continuous/node12/samples-test.cfg
.kokoro/continuous/node12/system-test.cfg
.kokoro/continuous/node12/test.cfg
.kokoro/docs.sh
.kokoro/lint.sh
.kokoro/populate-secrets.sh
.kokoro/presubmit/node10/common.cfg
.kokoro/presubmit/node12/common.cfg
.kokoro/presubmit/node12/samples-test.cfg
.kokoro/presubmit/node12/system-test.cfg
.kokoro/presubmit/node12/test.cfg
.kokoro/publish.sh
.kokoro/release/docs-devsite.cfg
.kokoro/release/docs-devsite.sh
.kokoro/release/docs.cfg
.kokoro/release/docs.sh
.kokoro/release/publish.cfg
.kokoro/samples-test.sh
.kokoro/system-test.sh
.kokoro/test.bat
.kokoro/test.sh
.kokoro/trampoline.sh
.kokoro/trampoline_v2.sh
.mocharc.js
.nycrc
.prettierignore
.prettierrc.js
.trampolinerc
CODE_OF_CONDUCT.md
CONTRIBUTING.md
LICENSE
README.md
api-extractor.json
renovate.json
samples/README.md
2020-11-20 03:10:20,570 synthtool [DEBUG] > Installing dependencies...
DEBUG:synthtool:Installing dependencies...
npm WARN deprecated ts-simple-ast@12.4.0: NOTICE: ts-simple-ast has been renamed to ts-morph and version reset to 1.0.0. Switch at your leisure...
npm WARN deprecated chokidar@2.1.8: Chokidar 2 will break on node v14+. Upgrade to chokidar 3 with 15x less dependencies.
npm WARN deprecated opn@6.0.0: The package has been renamed to `open`
npm WARN deprecated fsevents@1.2.13: fsevents 1 will break on node v14+ and could be using insecure binaries. Upgrade to fsevents 2.
npm WARN deprecated core-js@2.6.11: core-js@<3 is no longer maintained and not recommended for usage due to the number of issues. Please, upgrade your dependencies to the actual version of core-js@3.
npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated

> core-js@2.6.11 postinstall /home/kbuilder/.cache/synthtool/nodejs-firestore-session/node_modules/core-js
> node -e "try{require('./postinstall')}catch(e){}"

[96mThank you for using core-js ([94m https://github.com/zloirock/core-js [96m) for polyfilling JavaScript standard library![0m

[96mThe project needs your help! Please consider supporting of core-js on Open Collective or Patreon: [0m
[96m>[94m https://opencollective.com/core-js [0m
[96m>[94m https://www.patreon.com/zloirock [0m

[96mAlso, the author of core-js ([94m https://github.com/zloirock [96m) is looking for a good job -)[0m


> protobufjs@6.10.2 postinstall /home/kbuilder/.cache/synthtool/nodejs-firestore-session/node_modules/protobufjs
> node scripts/postinstall


> @compodoc/compodoc@1.1.11 postinstall /home/kbuilder/.cache/synthtool/nodejs-firestore-session/node_modules/@compodoc/compodoc
> opencollective-postinstall || exit 0

[96m[1mThank you for using @compodoc/compodoc![96m[1m
[0m[96mIf you rely on this package, please consider supporting our open collective:[22m[39m
> [94mhttps://opencollective.com/compodoc/donate[0m


> @google-cloud/connect-firestore@2.0.0 prepare /home/kbuilder/.cache/synthtool/nodejs-firestore-session
> npm run compile


> @google-cloud/connect-firestore@2.0.0 precompile /home/kbuilder/.cache/synthtool/nodejs-firestore-session
> gts clean

version: 14
Removing build ...

> @google-cloud/connect-firestore@2.0.0 compile /home/kbuilder/.cache/synthtool/nodejs-firestore-session
> tsc -p .

[96msrc/index.ts[0m:[93m27[0m:[93m11[0m - [91merror[0m[90m TS2559: [0mType 'StoreOptions' has no properties in common with type 'EventEmitterOptions'.

[7m27[0m     super(options || {});
[7m  [0m [91m          ~~~~~~~~~~~~~[0m

[96msrc/index.ts[0m:[93m37[0m:[93m54[0m - [91merror[0m[90m TS2694: [0mNamespace 'global.Express' has no exported member 'SessionData'.

[7m37[0m     callback: (err?: Error | null, session?: Express.SessionData) => void
[7m  [0m [91m                                                     ~~~~~~~~~~~[0m

[96msrc/index.ts[0m:[93m59[0m:[93m22[0m - [91merror[0m[90m TS2694: [0mNamespace 'global.Express' has no exported member 'SessionData'.

[7m59[0m     session: Express.SessionData,
[7m  [0m [91m                     ~~~~~~~~~~~[0m

[96msystem-test/session.ts[0m:[93m34[0m:[93m57[0m - [91merror[0m[90m TS2694: [0mNamespace 'global.Express' has no exported member 'SessionData'.

[7m34[0m     const sessionData = ({foo: 'bar'} as {}) as Express.SessionData;
[7m  [0m [91m                                                        ~~~~~~~~~~~[0m

[96mtest/test.ts[0m:[93m99[0m:[93m42[0m - [91merror[0m[90m TS2694: [0mNamespace 'global.Express' has no exported member 'SessionData'.

[7m99[0m     store.set(expectedSid, {} as Express.SessionData, assert.ifError);
[7m  [0m [91m                                         ~~~~~~~~~~~[0m


Found 5 errors.

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @google-cloud/connect-firestore@2.0.0 compile: `tsc -p .`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @google-cloud/connect-firestore@2.0.0 compile script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/kbuilder/.npm/_logs/2020-11-20T11_10_45_685Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @google-cloud/connect-firestore@2.0.0 prepare: `npm run compile`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @google-cloud/connect-firestore@2.0.0 prepare script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/kbuilder/.npm/_logs/2020-11-20T11_10_45_775Z-debug.log
2020-11-20 03:10:45,818 synthtool [ERROR] > Failed executing npm install:

None
ERROR:synthtool:Failed executing npm install:

None
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/nodejs-firestore-session/synth.py", line 15, in <module>
    node.install()
  File "/tmpfs/src/github/synthtool/synthtool/languages/node.py", line 167, in install
    shell.run(["npm", "install"], hide_output=hide_output)
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 39, in run
    raise exc
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 33, in run
    encoding="utf-8",
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['npm', 'install']' returned non-zero exit status 1.
2020-11-20 03:10:45,980 autosynth [ERROR] > Synthesis failed
2020-11-20 03:10:45,980 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at b623c57 build(node): add KOKORO_BUILD_ARTIFACTS_SUBDIR to env (#147)
2020-11-20 03:10:45,988 autosynth [DEBUG] > Running: git checkout autosynth
Switched to branch 'autosynth'
2020-11-20 03:10:45,994 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Removing node_modules/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 354, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 189, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 334, in _inner_main
    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 65, in synthesize_loop
    has_changes = toolbox.synthesize_version_in_new_branch(synthesizer, youngest)
  File "/tmpfs/src/github/synthtool/autosynth/synth_toolbox.py", line 259, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/feaaaf89-f535-452e-81ba-4578ca5d2c68/targets/github%2Fsynthtool;config=default/tests;query=nodejs-firestore-session;failed=false).

Autosynth is still having trouble generating nodejs-firestore-session. :sob:

Here's the output from running `synth.py`:

```
s
.github/ISSUE_TEMPLATE/bug_report.md
.github/ISSUE_TEMPLATE/feature_request.md
.github/ISSUE_TEMPLATE/support_request.md
.github/PULL_REQUEST_TEMPLATE.md
.github/release-please.yml
.github/workflows/ci.yaml
.kokoro/.gitattributes
.kokoro/common.cfg
.kokoro/continuous/node10/common.cfg
.kokoro/continuous/node10/docs.cfg
.kokoro/continuous/node10/test.cfg
.kokoro/continuous/node12/common.cfg
.kokoro/continuous/node12/lint.cfg
.kokoro/continuous/node12/samples-test.cfg
.kokoro/continuous/node12/system-test.cfg
.kokoro/continuous/node12/test.cfg
.kokoro/docs.sh
.kokoro/lint.sh
.kokoro/populate-secrets.sh
.kokoro/presubmit/node10/common.cfg
.kokoro/presubmit/node12/common.cfg
.kokoro/presubmit/node12/samples-test.cfg
.kokoro/presubmit/node12/system-test.cfg
.kokoro/presubmit/node12/test.cfg
.kokoro/publish.sh
.kokoro/release/docs-devsite.cfg
.kokoro/release/docs-devsite.sh
.kokoro/release/docs.cfg
.kokoro/release/docs.sh
.kokoro/release/publish.cfg
.kokoro/samples-test.sh
.kokoro/system-test.sh
.kokoro/test.bat
.kokoro/test.sh
.kokoro/trampoline.sh
.kokoro/trampoline_v2.sh
.mocharc.js
.nycrc
.prettierignore
.prettierrc.js
.trampolinerc
CODE_OF_CONDUCT.md
CONTRIBUTING.md
LICENSE
README.md
api-extractor.json
renovate.json
samples/README.md
2020-11-21 03:00:29,776 synthtool [DEBUG] > Installing dependencies...
DEBUG:synthtool:Installing dependencies...
npm WARN deprecated ts-simple-ast@12.4.0: NOTICE: ts-simple-ast has been renamed to ts-morph and version reset to 1.0.0. Switch at your leisure...
npm WARN deprecated chokidar@2.1.8: Chokidar 2 will break on node v14+. Upgrade to chokidar 3 with 15x less dependencies.
npm WARN deprecated opn@6.0.0: The package has been renamed to `open`
npm WARN deprecated fsevents@1.2.13: fsevents 1 will break on node v14+ and could be using insecure binaries. Upgrade to fsevents 2.
npm WARN deprecated core-js@2.6.11: core-js@<3 is no longer maintained and not recommended for usage due to the number of issues. Please, upgrade your dependencies to the actual version of core-js@3.
npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated

> core-js@2.6.11 postinstall /home/kbuilder/.cache/synthtool/nodejs-firestore-session/node_modules/core-js
> node -e "try{require('./postinstall')}catch(e){}"

[96mThank you for using core-js ([94m https://github.com/zloirock/core-js [96m) for polyfilling JavaScript standard library![0m

[96mThe project needs your help! Please consider supporting of core-js on Open Collective or Patreon: [0m
[96m>[94m https://opencollective.com/core-js [0m
[96m>[94m https://www.patreon.com/zloirock [0m

[96mAlso, the author of core-js ([94m https://github.com/zloirock [96m) is looking for a good job -)[0m


> protobufjs@6.10.2 postinstall /home/kbuilder/.cache/synthtool/nodejs-firestore-session/node_modules/protobufjs
> node scripts/postinstall


> @compodoc/compodoc@1.1.11 postinstall /home/kbuilder/.cache/synthtool/nodejs-firestore-session/node_modules/@compodoc/compodoc
> opencollective-postinstall || exit 0

[96m[1mThank you for using @compodoc/compodoc![96m[1m
[0m[96mIf you rely on this package, please consider supporting our open collective:[22m[39m
> [94mhttps://opencollective.com/compodoc/donate[0m


> @google-cloud/connect-firestore@2.0.0 prepare /home/kbuilder/.cache/synthtool/nodejs-firestore-session
> npm run compile


> @google-cloud/connect-firestore@2.0.0 precompile /home/kbuilder/.cache/synthtool/nodejs-firestore-session
> gts clean

version: 14
Removing build ...

> @google-cloud/connect-firestore@2.0.0 compile /home/kbuilder/.cache/synthtool/nodejs-firestore-session
> tsc -p .

[96msrc/index.ts[0m:[93m27[0m:[93m11[0m - [91merror[0m[90m TS2559: [0mType 'StoreOptions' has no properties in common with type 'EventEmitterOptions'.

[7m27[0m     super(options || {});
[7m  [0m [91m          ~~~~~~~~~~~~~[0m

[96msrc/index.ts[0m:[93m37[0m:[93m54[0m - [91merror[0m[90m TS2694: [0mNamespace 'global.Express' has no exported member 'SessionData'.

[7m37[0m     callback: (err?: Error | null, session?: Express.SessionData) => void
[7m  [0m [91m                                                     ~~~~~~~~~~~[0m

[96msrc/index.ts[0m:[93m59[0m:[93m22[0m - [91merror[0m[90m TS2694: [0mNamespace 'global.Express' has no exported member 'SessionData'.

[7m59[0m     session: Express.SessionData,
[7m  [0m [91m                     ~~~~~~~~~~~[0m

[96msystem-test/session.ts[0m:[93m34[0m:[93m57[0m - [91merror[0m[90m TS2694: [0mNamespace 'global.Express' has no exported member 'SessionData'.

[7m34[0m     const sessionData = ({foo: 'bar'} as {}) as Express.SessionData;
[7m  [0m [91m                                                        ~~~~~~~~~~~[0m

[96mtest/test.ts[0m:[93m99[0m:[93m42[0m - [91merror[0m[90m TS2694: [0mNamespace 'global.Express' has no exported member 'SessionData'.

[7m99[0m     store.set(expectedSid, {} as Express.SessionData, assert.ifError);
[7m  [0m [91m                                         ~~~~~~~~~~~[0m


Found 5 errors.

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @google-cloud/connect-firestore@2.0.0 compile: `tsc -p .`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @google-cloud/connect-firestore@2.0.0 compile script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/kbuilder/.npm/_logs/2020-11-21T11_00_51_388Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @google-cloud/connect-firestore@2.0.0 prepare: `npm run compile`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @google-cloud/connect-firestore@2.0.0 prepare script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/kbuilder/.npm/_logs/2020-11-21T11_00_51_468Z-debug.log
2020-11-21 03:00:51,509 synthtool [ERROR] > Failed executing npm install:

None
ERROR:synthtool:Failed executing npm install:

None
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/nodejs-firestore-session/synth.py", line 15, in <module>
    node.install()
  File "/tmpfs/src/github/synthtool/synthtool/languages/node.py", line 167, in install
    shell.run(["npm", "install"], hide_output=hide_output)
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 39, in run
    raise exc
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 33, in run
    encoding="utf-8",
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['npm', 'install']' returned non-zero exit status 1.
2020-11-21 03:00:51,652 autosynth [ERROR] > Synthesis failed
2020-11-21 03:00:51,652 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at b623c57 build(node): add KOKORO_BUILD_ARTIFACTS_SUBDIR to env (#147)
2020-11-21 03:00:51,659 autosynth [DEBUG] > Running: git checkout autosynth
Switched to branch 'autosynth'
2020-11-21 03:00:51,663 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Removing node_modules/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 354, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 189, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 334, in _inner_main
    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 65, in synthesize_loop
    has_changes = toolbox.synthesize_version_in_new_branch(synthesizer, youngest)
  File "/tmpfs/src/github/synthtool/autosynth/synth_toolbox.py", line 259, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/b41603b7-0ed3-4691-8e18-4471e91b558b/targets/github%2Fsynthtool;config=default/tests;query=nodejs-firestore-session;failed=false).

Autosynth is still having trouble generating nodejs-firestore-session. :sob:

Here's the output from running `synth.py`:

```
s
.github/ISSUE_TEMPLATE/bug_report.md
.github/ISSUE_TEMPLATE/feature_request.md
.github/ISSUE_TEMPLATE/support_request.md
.github/PULL_REQUEST_TEMPLATE.md
.github/release-please.yml
.github/workflows/ci.yaml
.kokoro/.gitattributes
.kokoro/common.cfg
.kokoro/continuous/node10/common.cfg
.kokoro/continuous/node10/docs.cfg
.kokoro/continuous/node10/test.cfg
.kokoro/continuous/node12/common.cfg
.kokoro/continuous/node12/lint.cfg
.kokoro/continuous/node12/samples-test.cfg
.kokoro/continuous/node12/system-test.cfg
.kokoro/continuous/node12/test.cfg
.kokoro/docs.sh
.kokoro/lint.sh
.kokoro/populate-secrets.sh
.kokoro/presubmit/node10/common.cfg
.kokoro/presubmit/node12/common.cfg
.kokoro/presubmit/node12/samples-test.cfg
.kokoro/presubmit/node12/system-test.cfg
.kokoro/presubmit/node12/test.cfg
.kokoro/publish.sh
.kokoro/release/docs-devsite.cfg
.kokoro/release/docs-devsite.sh
.kokoro/release/docs.cfg
.kokoro/release/docs.sh
.kokoro/release/publish.cfg
.kokoro/samples-test.sh
.kokoro/system-test.sh
.kokoro/test.bat
.kokoro/test.sh
.kokoro/trampoline.sh
.kokoro/trampoline_v2.sh
.mocharc.js
.nycrc
.prettierignore
.prettierrc.js
.trampolinerc
CODE_OF_CONDUCT.md
CONTRIBUTING.md
LICENSE
README.md
api-extractor.json
renovate.json
samples/README.md
2020-11-22 03:08:52,671 synthtool [DEBUG] > Installing dependencies...
DEBUG:synthtool:Installing dependencies...
npm WARN deprecated ts-simple-ast@12.4.0: NOTICE: ts-simple-ast has been renamed to ts-morph and version reset to 1.0.0. Switch at your leisure...
npm WARN deprecated chokidar@2.1.8: Chokidar 2 will break on node v14+. Upgrade to chokidar 3 with 15x less dependencies.
npm WARN deprecated opn@6.0.0: The package has been renamed to `open`
npm WARN deprecated fsevents@1.2.13: fsevents 1 will break on node v14+ and could be using insecure binaries. Upgrade to fsevents 2.
npm WARN deprecated core-js@2.6.11: core-js@<3 is no longer maintained and not recommended for usage due to the number of issues. Please, upgrade your dependencies to the actual version of core-js@3.
npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated

> core-js@2.6.11 postinstall /home/kbuilder/.cache/synthtool/nodejs-firestore-session/node_modules/core-js
> node -e "try{require('./postinstall')}catch(e){}"

[96mThank you for using core-js ([94m https://github.com/zloirock/core-js [96m) for polyfilling JavaScript standard library![0m

[96mThe project needs your help! Please consider supporting of core-js on Open Collective or Patreon: [0m
[96m>[94m https://opencollective.com/core-js [0m
[96m>[94m https://www.patreon.com/zloirock [0m

[96mAlso, the author of core-js ([94m https://github.com/zloirock [96m) is looking for a good job -)[0m


> protobufjs@6.10.2 postinstall /home/kbuilder/.cache/synthtool/nodejs-firestore-session/node_modules/protobufjs
> node scripts/postinstall


> @compodoc/compodoc@1.1.11 postinstall /home/kbuilder/.cache/synthtool/nodejs-firestore-session/node_modules/@compodoc/compodoc
> opencollective-postinstall || exit 0

[96m[1mThank you for using @compodoc/compodoc![96m[1m
[0m[96mIf you rely on this package, please consider supporting our open collective:[22m[39m
> [94mhttps://opencollective.com/compodoc/donate[0m


> @google-cloud/connect-firestore@2.0.0 prepare /home/kbuilder/.cache/synthtool/nodejs-firestore-session
> npm run compile


> @google-cloud/connect-firestore@2.0.0 precompile /home/kbuilder/.cache/synthtool/nodejs-firestore-session
> gts clean

version: 14
Removing build ...

> @google-cloud/connect-firestore@2.0.0 compile /home/kbuilder/.cache/synthtool/nodejs-firestore-session
> tsc -p .

[96msrc/index.ts[0m:[93m27[0m:[93m11[0m - [91merror[0m[90m TS2559: [0mType 'StoreOptions' has no properties in common with type 'EventEmitterOptions'.

[7m27[0m     super(options || {});
[7m  [0m [91m          ~~~~~~~~~~~~~[0m

[96msrc/index.ts[0m:[93m37[0m:[93m54[0m - [91merror[0m[90m TS2694: [0mNamespace 'global.Express' has no exported member 'SessionData'.

[7m37[0m     callback: (err?: Error | null, session?: Express.SessionData) => void
[7m  [0m [91m                                                     ~~~~~~~~~~~[0m

[96msrc/index.ts[0m:[93m59[0m:[93m22[0m - [91merror[0m[90m TS2694: [0mNamespace 'global.Express' has no exported member 'SessionData'.

[7m59[0m     session: Express.SessionData,
[7m  [0m [91m                     ~~~~~~~~~~~[0m

[96msystem-test/session.ts[0m:[93m34[0m:[93m57[0m - [91merror[0m[90m TS2694: [0mNamespace 'global.Express' has no exported member 'SessionData'.

[7m34[0m     const sessionData = ({foo: 'bar'} as {}) as Express.SessionData;
[7m  [0m [91m                                                        ~~~~~~~~~~~[0m

[96mtest/test.ts[0m:[93m99[0m:[93m42[0m - [91merror[0m[90m TS2694: [0mNamespace 'global.Express' has no exported member 'SessionData'.

[7m99[0m     store.set(expectedSid, {} as Express.SessionData, assert.ifError);
[7m  [0m [91m                                         ~~~~~~~~~~~[0m


Found 5 errors.

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @google-cloud/connect-firestore@2.0.0 compile: `tsc -p .`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @google-cloud/connect-firestore@2.0.0 compile script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/kbuilder/.npm/_logs/2020-11-22T11_09_21_963Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @google-cloud/connect-firestore@2.0.0 prepare: `npm run compile`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @google-cloud/connect-firestore@2.0.0 prepare script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/kbuilder/.npm/_logs/2020-11-22T11_09_22_062Z-debug.log
2020-11-22 03:09:22,102 synthtool [ERROR] > Failed executing npm install:

None
ERROR:synthtool:Failed executing npm install:

None
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/nodejs-firestore-session/synth.py", line 15, in <module>
    node.install()
  File "/tmpfs/src/github/synthtool/synthtool/languages/node.py", line 167, in install
    shell.run(["npm", "install"], hide_output=hide_output)
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 39, in run
    raise exc
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 33, in run
    encoding="utf-8",
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['npm', 'install']' returned non-zero exit status 1.
2020-11-22 03:09:22,548 autosynth [ERROR] > Synthesis failed
2020-11-22 03:09:22,548 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at b623c57 build(node): add KOKORO_BUILD_ARTIFACTS_SUBDIR to env (#147)
2020-11-22 03:09:22,556 autosynth [DEBUG] > Running: git checkout autosynth
Switched to branch 'autosynth'
2020-11-22 03:09:22,561 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Removing node_modules/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 354, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 189, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 334, in _inner_main
    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 65, in synthesize_loop
    has_changes = toolbox.synthesize_version_in_new_branch(synthesizer, youngest)
  File "/tmpfs/src/github/synthtool/autosynth/synth_toolbox.py", line 259, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/386143a8-9a1e-43b1-a906-a34d08b9bc98/targets/github%2Fsynthtool;config=default/tests;query=nodejs-firestore-session;failed=false).

Autosynth is still having trouble generating nodejs-firestore-session. :sob:

Here's the output from running `synth.py`:

```
s
.github/ISSUE_TEMPLATE/bug_report.md
.github/ISSUE_TEMPLATE/feature_request.md
.github/ISSUE_TEMPLATE/support_request.md
.github/PULL_REQUEST_TEMPLATE.md
.github/release-please.yml
.github/workflows/ci.yaml
.kokoro/.gitattributes
.kokoro/common.cfg
.kokoro/continuous/node10/common.cfg
.kokoro/continuous/node10/docs.cfg
.kokoro/continuous/node10/test.cfg
.kokoro/continuous/node12/common.cfg
.kokoro/continuous/node12/lint.cfg
.kokoro/continuous/node12/samples-test.cfg
.kokoro/continuous/node12/system-test.cfg
.kokoro/continuous/node12/test.cfg
.kokoro/docs.sh
.kokoro/lint.sh
.kokoro/populate-secrets.sh
.kokoro/presubmit/node10/common.cfg
.kokoro/presubmit/node12/common.cfg
.kokoro/presubmit/node12/samples-test.cfg
.kokoro/presubmit/node12/system-test.cfg
.kokoro/presubmit/node12/test.cfg
.kokoro/publish.sh
.kokoro/release/docs-devsite.cfg
.kokoro/release/docs-devsite.sh
.kokoro/release/docs.cfg
.kokoro/release/docs.sh
.kokoro/release/publish.cfg
.kokoro/samples-test.sh
.kokoro/system-test.sh
.kokoro/test.bat
.kokoro/test.sh
.kokoro/trampoline.sh
.kokoro/trampoline_v2.sh
.mocharc.js
.nycrc
.prettierignore
.prettierrc.js
.trampolinerc
CODE_OF_CONDUCT.md
CONTRIBUTING.md
LICENSE
README.md
api-extractor.json
renovate.json
samples/README.md
2020-11-23 03:09:10,064 synthtool [DEBUG] > Installing dependencies...
DEBUG:synthtool:Installing dependencies...
npm WARN deprecated ts-simple-ast@12.4.0: NOTICE: ts-simple-ast has been renamed to ts-morph and version reset to 1.0.0. Switch at your leisure...
npm WARN deprecated chokidar@2.1.8: Chokidar 2 will break on node v14+. Upgrade to chokidar 3 with 15x less dependencies.
npm WARN deprecated opn@6.0.0: The package has been renamed to `open`
npm WARN deprecated fsevents@1.2.13: fsevents 1 will break on node v14+ and could be using insecure binaries. Upgrade to fsevents 2.
npm WARN deprecated core-js@2.6.11: core-js@<3 is no longer maintained and not recommended for usage due to the number of issues. Please, upgrade your dependencies to the actual version of core-js@3.
npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated

> core-js@2.6.11 postinstall /home/kbuilder/.cache/synthtool/nodejs-firestore-session/node_modules/core-js
> node -e "try{require('./postinstall')}catch(e){}"

[96mThank you for using core-js ([94m https://github.com/zloirock/core-js [96m) for polyfilling JavaScript standard library![0m

[96mThe project needs your help! Please consider supporting of core-js on Open Collective or Patreon: [0m
[96m>[94m https://opencollective.com/core-js [0m
[96m>[94m https://www.patreon.com/zloirock [0m

[96mAlso, the author of core-js ([94m https://github.com/zloirock [96m) is looking for a good job -)[0m


> protobufjs@6.10.2 postinstall /home/kbuilder/.cache/synthtool/nodejs-firestore-session/node_modules/protobufjs
> node scripts/postinstall


> @compodoc/compodoc@1.1.11 postinstall /home/kbuilder/.cache/synthtool/nodejs-firestore-session/node_modules/@compodoc/compodoc
> opencollective-postinstall || exit 0

[96m[1mThank you for using @compodoc/compodoc![96m[1m
[0m[96mIf you rely on this package, please consider supporting our open collective:[22m[39m
> [94mhttps://opencollective.com/compodoc/donate[0m


> @google-cloud/connect-firestore@2.0.0 prepare /home/kbuilder/.cache/synthtool/nodejs-firestore-session
> npm run compile


> @google-cloud/connect-firestore@2.0.0 precompile /home/kbuilder/.cache/synthtool/nodejs-firestore-session
> gts clean

version: 14
Removing build ...

> @google-cloud/connect-firestore@2.0.0 compile /home/kbuilder/.cache/synthtool/nodejs-firestore-session
> tsc -p .

[96msrc/index.ts[0m:[93m27[0m:[93m11[0m - [91merror[0m[90m TS2559: [0mType 'StoreOptions' has no properties in common with type 'EventEmitterOptions'.

[7m27[0m     super(options || {});
[7m  [0m [91m          ~~~~~~~~~~~~~[0m

[96msrc/index.ts[0m:[93m37[0m:[93m54[0m - [91merror[0m[90m TS2694: [0mNamespace 'global.Express' has no exported member 'SessionData'.

[7m37[0m     callback: (err?: Error | null, session?: Express.SessionData) => void
[7m  [0m [91m                                                     ~~~~~~~~~~~[0m

[96msrc/index.ts[0m:[93m59[0m:[93m22[0m - [91merror[0m[90m TS2694: [0mNamespace 'global.Express' has no exported member 'SessionData'.

[7m59[0m     session: Express.SessionData,
[7m  [0m [91m                     ~~~~~~~~~~~[0m

[96msystem-test/session.ts[0m:[93m34[0m:[93m57[0m - [91merror[0m[90m TS2694: [0mNamespace 'global.Express' has no exported member 'SessionData'.

[7m34[0m     const sessionData = ({foo: 'bar'} as {}) as Express.SessionData;
[7m  [0m [91m                                                        ~~~~~~~~~~~[0m

[96mtest/test.ts[0m:[93m99[0m:[93m42[0m - [91merror[0m[90m TS2694: [0mNamespace 'global.Express' has no exported member 'SessionData'.

[7m99[0m     store.set(expectedSid, {} as Express.SessionData, assert.ifError);
[7m  [0m [91m                                         ~~~~~~~~~~~[0m


Found 5 errors.

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @google-cloud/connect-firestore@2.0.0 compile: `tsc -p .`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @google-cloud/connect-firestore@2.0.0 compile script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/kbuilder/.npm/_logs/2020-11-23T11_09_39_338Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @google-cloud/connect-firestore@2.0.0 prepare: `npm run compile`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @google-cloud/connect-firestore@2.0.0 prepare script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/kbuilder/.npm/_logs/2020-11-23T11_09_39_434Z-debug.log
2020-11-23 03:09:39,465 synthtool [ERROR] > Failed executing npm install:

None
ERROR:synthtool:Failed executing npm install:

None
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/nodejs-firestore-session/synth.py", line 15, in <module>
    node.install()
  File "/tmpfs/src/github/synthtool/synthtool/languages/node.py", line 167, in install
    shell.run(["npm", "install"], hide_output=hide_output)
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 39, in run
    raise exc
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 33, in run
    encoding="utf-8",
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['npm', 'install']' returned non-zero exit status 1.
2020-11-23 03:09:39,569 autosynth [ERROR] > Synthesis failed
2020-11-23 03:09:39,570 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at b623c57 build(node): add KOKORO_BUILD_ARTIFACTS_SUBDIR to env (#147)
2020-11-23 03:09:39,578 autosynth [DEBUG] > Running: git checkout autosynth
Switched to branch 'autosynth'
2020-11-23 03:09:39,583 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Removing node_modules/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 354, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 189, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 334, in _inner_main
    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 65, in synthesize_loop
    has_changes = toolbox.synthesize_version_in_new_branch(synthesizer, youngest)
  File "/tmpfs/src/github/synthtool/autosynth/synth_toolbox.py", line 259, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/ddae8a7f-1e14-4418-b4b8-bfb0d012c0e4/targets/github%2Fsynthtool;config=default/tests;query=nodejs-firestore-session;failed=false).

Autosynth is still having trouble generating nodejs-firestore-session. :sob:

Here's the output from running `synth.py`:

```
s
.github/ISSUE_TEMPLATE/bug_report.md
.github/ISSUE_TEMPLATE/feature_request.md
.github/ISSUE_TEMPLATE/support_request.md
.github/PULL_REQUEST_TEMPLATE.md
.github/release-please.yml
.github/workflows/ci.yaml
.kokoro/.gitattributes
.kokoro/common.cfg
.kokoro/continuous/node10/common.cfg
.kokoro/continuous/node10/docs.cfg
.kokoro/continuous/node10/test.cfg
.kokoro/continuous/node12/common.cfg
.kokoro/continuous/node12/lint.cfg
.kokoro/continuous/node12/samples-test.cfg
.kokoro/continuous/node12/system-test.cfg
.kokoro/continuous/node12/test.cfg
.kokoro/docs.sh
.kokoro/lint.sh
.kokoro/populate-secrets.sh
.kokoro/presubmit/node10/common.cfg
.kokoro/presubmit/node12/common.cfg
.kokoro/presubmit/node12/samples-test.cfg
.kokoro/presubmit/node12/system-test.cfg
.kokoro/presubmit/node12/test.cfg
.kokoro/publish.sh
.kokoro/release/docs-devsite.cfg
.kokoro/release/docs-devsite.sh
.kokoro/release/docs.cfg
.kokoro/release/docs.sh
.kokoro/release/publish.cfg
.kokoro/samples-test.sh
.kokoro/system-test.sh
.kokoro/test.bat
.kokoro/test.sh
.kokoro/trampoline.sh
.kokoro/trampoline_v2.sh
.mocharc.js
.nycrc
.prettierignore
.prettierrc.js
.trampolinerc
CODE_OF_CONDUCT.md
CONTRIBUTING.md
LICENSE
README.md
api-extractor.json
renovate.json
samples/README.md
2020-11-24 02:57:40,370 synthtool [DEBUG] > Installing dependencies...
DEBUG:synthtool:Installing dependencies...
npm WARN deprecated ts-simple-ast@12.4.0: NOTICE: ts-simple-ast has been renamed to ts-morph and version reset to 1.0.0. Switch at your leisure...
npm WARN deprecated chokidar@2.1.8: Chokidar 2 will break on node v14+. Upgrade to chokidar 3 with 15x less dependencies.
npm WARN deprecated opn@6.0.0: The package has been renamed to `open`
npm WARN deprecated fsevents@1.2.13: fsevents 1 will break on node v14+ and could be using insecure binaries. Upgrade to fsevents 2.
npm WARN deprecated core-js@2.6.11: core-js@<3 is no longer maintained and not recommended for usage due to the number of issues. Please, upgrade your dependencies to the actual version of core-js@3.
npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated

> core-js@2.6.11 postinstall /home/kbuilder/.cache/synthtool/nodejs-firestore-session/node_modules/core-js
> node -e "try{require('./postinstall')}catch(e){}"

[96mThank you for using core-js ([94m https://github.com/zloirock/core-js [96m) for polyfilling JavaScript standard library![0m

[96mThe project needs your help! Please consider supporting of core-js on Open Collective or Patreon: [0m
[96m>[94m https://opencollective.com/core-js [0m
[96m>[94m https://www.patreon.com/zloirock [0m

[96mAlso, the author of core-js ([94m https://github.com/zloirock [96m) is looking for a good job -)[0m


> protobufjs@6.10.2 postinstall /home/kbuilder/.cache/synthtool/nodejs-firestore-session/node_modules/protobufjs
> node scripts/postinstall


> @compodoc/compodoc@1.1.11 postinstall /home/kbuilder/.cache/synthtool/nodejs-firestore-session/node_modules/@compodoc/compodoc
> opencollective-postinstall || exit 0

[96m[1mThank you for using @compodoc/compodoc![96m[1m
[0m[96mIf you rely on this package, please consider supporting our open collective:[22m[39m
> [94mhttps://opencollective.com/compodoc/donate[0m


> @google-cloud/connect-firestore@2.0.0 prepare /home/kbuilder/.cache/synthtool/nodejs-firestore-session
> npm run compile


> @google-cloud/connect-firestore@2.0.0 precompile /home/kbuilder/.cache/synthtool/nodejs-firestore-session
> gts clean

version: 14
Removing build ...

> @google-cloud/connect-firestore@2.0.0 compile /home/kbuilder/.cache/synthtool/nodejs-firestore-session
> tsc -p .

[96msrc/index.ts[0m:[93m27[0m:[93m11[0m - [91merror[0m[90m TS2559: [0mType 'StoreOptions' has no properties in common with type 'EventEmitterOptions'.

[7m27[0m     super(options || {});
[7m  [0m [91m          ~~~~~~~~~~~~~[0m

[96msrc/index.ts[0m:[93m37[0m:[93m54[0m - [91merror[0m[90m TS2694: [0mNamespace 'global.Express' has no exported member 'SessionData'.

[7m37[0m     callback: (err?: Error | null, session?: Express.SessionData) => void
[7m  [0m [91m                                                     ~~~~~~~~~~~[0m

[96msrc/index.ts[0m:[93m59[0m:[93m22[0m - [91merror[0m[90m TS2694: [0mNamespace 'global.Express' has no exported member 'SessionData'.

[7m59[0m     session: Express.SessionData,
[7m  [0m [91m                     ~~~~~~~~~~~[0m

[96msystem-test/session.ts[0m:[93m34[0m:[93m57[0m - [91merror[0m[90m TS2694: [0mNamespace 'global.Express' has no exported member 'SessionData'.

[7m34[0m     const sessionData = ({foo: 'bar'} as {}) as Express.SessionData;
[7m  [0m [91m                                                        ~~~~~~~~~~~[0m

[96mtest/test.ts[0m:[93m99[0m:[93m42[0m - [91merror[0m[90m TS2694: [0mNamespace 'global.Express' has no exported member 'SessionData'.

[7m99[0m     store.set(expectedSid, {} as Express.SessionData, assert.ifError);
[7m  [0m [91m                                         ~~~~~~~~~~~[0m


Found 5 errors.

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @google-cloud/connect-firestore@2.0.0 compile: `tsc -p .`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @google-cloud/connect-firestore@2.0.0 compile script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/kbuilder/.npm/_logs/2020-11-24T10_58_05_430Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @google-cloud/connect-firestore@2.0.0 prepare: `npm run compile`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @google-cloud/connect-firestore@2.0.0 prepare script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/kbuilder/.npm/_logs/2020-11-24T10_58_05_490Z-debug.log
2020-11-24 02:58:05,535 synthtool [ERROR] > Failed executing npm install:

None
ERROR:synthtool:Failed executing npm install:

None
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/nodejs-firestore-session/synth.py", line 15, in <module>
    node.install()
  File "/tmpfs/src/github/synthtool/synthtool/languages/node.py", line 167, in install
    shell.run(["npm", "install"], hide_output=hide_output)
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 39, in run
    raise exc
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 33, in run
    encoding="utf-8",
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['npm', 'install']' returned non-zero exit status 1.
2020-11-24 02:58:05,651 autosynth [ERROR] > Synthesis failed
2020-11-24 02:58:05,652 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at b623c57 build(node): add KOKORO_BUILD_ARTIFACTS_SUBDIR to env (#147)
2020-11-24 02:58:05,660 autosynth [DEBUG] > Running: git checkout autosynth
Switched to branch 'autosynth'
2020-11-24 02:58:05,665 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Removing node_modules/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 354, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 189, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 334, in _inner_main
    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 65, in synthesize_loop
    has_changes = toolbox.synthesize_version_in_new_branch(synthesizer, youngest)
  File "/tmpfs/src/github/synthtool/autosynth/synth_toolbox.py", line 259, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/e156b84a-5b22-448d-9176-aaf231f23dd4/targets/github%2Fsynthtool;config=default/tests;query=nodejs-firestore-session;failed=false).

--- a/package.json
+++ b/package.json
@@ -43,7 +43,6 @@
     "@google-cloud/firestore": "^4.0.0",
     "@microsoft/api-documenter": "^7.8.10",
     "@microsoft/api-extractor": "^7.8.10",
-    "@types/express-session": "^1.17.3",
     "@types/mocha": "^8.0.0",
     "c8": "^7.0.0",
     "express": "^4.16.4",

--- a/package.json
+++ b/package.json
@@ -43,6 +43,7 @@
     "@google-cloud/firestore": "^4.0.0",
     "@microsoft/api-documenter": "^7.8.10",
     "@microsoft/api-extractor": "^7.8.10",
+    "@types/express-session": "^1.17.3",
     "@types/mocha": "^8.0.0",
     "c8": "^7.0.0",
     "express": "^4.16.4",

--- a/package.json
+++ b/package.json
@@ -41,16 +41,16 @@
   "devDependencies": {
     "@compodoc/compodoc": "^1.1.9",
     "@google-cloud/firestore": "^4.0.0",
-    "@types/express-session": "^1.15.12",
+    "@microsoft/api-documenter": "^7.8.10",
+    "@microsoft/api-extractor": "^7.8.10",
+    "@types/express-session": "^1.17.3",
     "@types/mocha": "^8.0.0",
+    "c8": "^7.0.0",
     "express": "^4.16.4",
     "express-session": "^1.15.6",
     "gts": "^2.0.0",
     "linkinator": "^2.0.0",
     "mocha": "^8.0.0",
-    "c8": "^7.0.0",
-    "typescript": "^3.8.3",
-    "@microsoft/api-documenter": "^7.8.10",
-    "@microsoft/api-extractor": "^7.8.10"
+    "typescript": "^3.8.3"
   }
 }

--- a/src/index.ts
+++ b/src/index.ts
@@ -24,8 +24,7 @@ export class FirestoreStore extends Store {
   db: Firestore;
   kind: string;
   constructor(options: StoreOptions) {
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    super((options || {}) as any);
+    super();
     this.db = options.dataset;
     if (!this.db) {
       throw new Error('No dataset provided to Firestore Session.');

--- a/src/index.ts
+++ b/src/index.ts
@@ -13,7 +13,7 @@
 // limitations under the License.
 
 import {Firestore} from '@google-cloud/firestore';
-import {Store} from 'express-session';
+import {Store, SessionData} from 'express-session';
 
 export interface StoreOptions {
   dataset: Firestore;
@@ -24,7 +24,8 @@ export class FirestoreStore extends Store {
   db: Firestore;
   kind: string;
   constructor(options: StoreOptions) {
-    super(options || {});
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    super((options || {}) as any);
     this.db = options.dataset;
     if (!this.db) {
       throw new Error('No dataset provided to Firestore Session.');
@@ -34,7 +35,7 @@ export class FirestoreStore extends Store {
 
   get = (
     sid: string,
-    callback: (err?: Error | null, session?: Express.SessionData) => void
+    callback: (err?: Error | null, session?: SessionData) => void
   ) => {
     this.db
       .collection(this.kind)
@@ -56,7 +57,7 @@ export class FirestoreStore extends Store {
 
   set = (
     sid: string,
-    session: Express.SessionData,
+    session: SessionData,
     callback?: (err?: Error) => void
   ) => {
     callback = callback || (() => {});

--- a/system-test/session.ts
+++ b/system-test/session.ts
@@ -16,6 +16,7 @@ import {Firestore} from '@google-cloud/firestore';
 import * as assert from 'assert';
 import {describe, it} from 'mocha';
 import {FirestoreStore} from '../src';
+import {SessionData} from 'express-session';
 
 const store = new FirestoreStore({
   dataset: new Firestore(),
@@ -31,7 +32,7 @@ describe('system tests', () => {
   });
 
   it('Should create and retrieve a session', done => {
-    const sessionData = ({foo: 'bar'} as {}) as Express.SessionData;
+    const sessionData = ({foo: 'bar'} as {}) as SessionData;
     store.set('123', sessionData, err => {
       assert.ifError(err);
       store.get('123', (err, session) => {

--- a/test/test.ts
+++ b/test/test.ts
@@ -17,6 +17,7 @@ import * as assert from 'assert';
 import {describe, it} from 'mocha';
 
 import {FirestoreStore} from '../src/index';
+import {SessionData} from 'express-session';
 
 describe('firestore session', () => {
   it('should throw without dataset', done => {
@@ -96,7 +97,7 @@ describe('firestore session', () => {
     });
 
     const expectedSid = 'sid';
-    store.set(expectedSid, {} as Express.SessionData, assert.ifError);
+    store.set(expectedSid, {} as SessionData, assert.ifError);
   });
 
   it('should destroy a document', done => {

